DIR 02d9c6a432721d5a9db1b87b75f67890
--- pr34176.c.{avr25}.{2}.{vanilla}.s	2020-08-14 10:08:50.725788596 +0000
+++ pr34176.c.{avr25}.{2}.{ccmode}.s	2020-08-14 10:08:53.593742927 +0000
@@ -53,22 +53,23 @@
 	rcall .
 	in r28,__SP_L__
 	in r29,__SP_H__
-.L12:
 /* prologue: function */
 /* frame size = 2 */
 /* stack size = 2 */
 .L__stack_usage = 2
 	movw r24,r28
 	adiw r24,1
+	movw r14,r24
 	rcall hash_find_entry
 	ldd r16,Y+1
 	ldd r17,Y+2
 	movw r24,r16
+	movw r12,r14
 .L9:
 	rcall foo
 	cp r16,__zero_reg__
 	cpc r17,__zero_reg__
-	breq .L12
+	breq .L7
 	lsl r16
 	rol r17
 	lsl r16
@@ -78,8 +79,7 @@
 	ror r14
 	asr r15
 	ror r14
-	movw r24,r28
-	adiw r24,1
+	movw r24,r12
 	rcall hash_find_entry
 	ldd r16,Y+1
 	ldd r17,Y+2
@@ -95,6 +95,13 @@
 	pop __tmp_reg__
 	pop __tmp_reg__
 	ret
+.L7:
+	movw r24,r12
+	rcall hash_find_entry
+	ldd r16,Y+1
+	ldd r17,Y+2
+	movw r24,r16
+	rjmp .L9
 	.size	main, .-main
 	.local	count.0
 	.comm	count.0,2,1
