DIR 8f5d78d2547230ad55877a9eb9e9fc3a
--- 920501-6.c.{atmega128}.{3}.{vanilla}.s	2020-08-05 18:51:12.885443146 +0000
+++ 920501-6.c.{atmega128}.{3}.{ccmode}.s	2020-08-05 18:51:12.589439037 +0000
@@ -79,8 +79,9 @@
 	movw r14,r22
 	mov r26,r24
 	mov r17,r25
-	ld r24,Y
+	movw r30,r28
 	adiw r28,1
+	ld r24,Z
 	mov r30,r24
 	mov __tmp_reg__,r24
 	lsl r0
@@ -408,10 +409,10 @@
 	breq .+4
 	brlo .+2
 	rjmp .L26
-	ldd r18,Y+26
-	ldd r19,Y+27
-	std Y+17,r19
-	std Y+16,r18
+	ldd r24,Y+26
+	ldd r25,Y+27
+	std Y+17,r25
+	std Y+16,r24
 .L25:
 	ldd r5,Y+8
 	ldd r14,Y+9
@@ -589,8 +590,8 @@
 	cpc r27,__zero_reg__
 	brsh .+2
 	rjmp .L21
-	ldi r20,lo8(3)
-	mov r10,r20
+	ldi r18,lo8(3)
+	mov r10,r18
 	mov r11,__zero_reg__
 	mov r12,__zero_reg__
 	mov r13,__zero_reg__
@@ -611,8 +612,8 @@
 	call __cmpdi2_s8
 	brne .+2
 	rjmp .L22
-	ldi r19,lo8(3)
-	mov r4,r19
+	ldi r25,lo8(3)
+	mov r4,r25
 	mov r5,__zero_reg__
 	mov r6,__zero_reg__
 	mov r7,__zero_reg__
@@ -662,29 +663,29 @@
 	adiw r26,1
 	st X,r31
 	sbiw r26,1
-	ldd r18,Y+10
+	ldd r24,Y+10
 	adiw r26,2
-	st X,r18
+	st X,r24
 	sbiw r26,2
-	ldd r19,Y+11
+	ldd r25,Y+11
 	adiw r26,3
-	st X,r19
+	st X,r25
 	sbiw r26,3
-	ldd r24,Y+12
+	ldd r30,Y+12
 	adiw r26,4
-	st X,r24
+	st X,r30
 	sbiw r26,4
-	ldd r25,Y+13
+	ldd r31,Y+13
 	adiw r26,5
-	st X,r25
+	st X,r31
 	sbiw r26,5
-	ldd r30,Y+14
+	ldd r24,Y+14
 	adiw r26,6
-	st X,r30
+	st X,r24
 	sbiw r26,6
-	ldd r31,Y+15
+	ldd r25,Y+15
 	adiw r26,7
-	st X,r31
+	st X,r25
 	sbiw r26,7
 	adiw r26,8
 	std Y+17,r27
@@ -727,18 +728,19 @@
 	call __cmpdi2
 	brlo .+2
 	rjmp .L25
-	ldd r24,Y+16
-	ldd r25,Y+17
-	ldd r18,Y+26
-	ldd r19,Y+27
-	sub r24,r18
-	sbc r25,r19
-	asr r25
-	ror r24
-	asr r25
-	ror r24
-	asr r25
-	ror r24
+	ldd r18,Y+16
+	ldd r19,Y+17
+	ldd r26,Y+26
+	ldd r27,Y+27
+	sub r18,r26
+	sbc r19,r27
+	asr r19
+	ror r18
+	asr r19
+	ror r18
+	asr r19
+	ror r18
+	movw r24,r18
 .L17:
 	ldd r26,Y+16
 	ldd r27,Y+17
@@ -790,12 +792,12 @@
 	pop r2
 	ret
 .L26:
-	ldd r24,Y+26
-	ldd r25,Y+27
-	std Y+17,r25
-	std Y+16,r24
-	ldi r25,0
+	ldd r30,Y+26
+	ldd r31,Y+27
+	std Y+17,r31
+	std Y+16,r30
 	ldi r24,0
+	ldi r25,0
 	rjmp .L17
 	.size	plist, .-plist
 	.section	.rodata.str1.1,"aMS",@progbits,1
@@ -835,13 +837,13 @@
 	adiw r28,90-63
 	std Y+63,__zero_reg__
 	sbiw r28,90-63
-	ldi r25,lo8(50)
-	mov r2,r25
+	ldi r18,lo8(50)
+	mov r2,r18
 	mov r3,__zero_reg__
-	ldi r18,lo8(.LC0+2)
-	mov r8,r18
-	ldi r18,hi8(.LC0+2)
-	mov r9,r18
+	ldi r19,lo8(.LC0+2)
+	mov r8,r19
+	ldi r19,hi8(.LC0+2)
+	mov r9,r19
 .L32:
 	adiw r28,81-63
 	ldd r18,Y+63
@@ -904,20 +906,21 @@
 	adiw r28,90-63
 	std Y+63,r25
 	sbiw r28,90-63
-	movw r30,r8
+	movw r24,r8
+	ldi r18,-1
+	sub r8,r18
+	sbc r9,r18
+	movw r30,r24
 	ld r24,Z
-	ldi r31,-1
-	sub r8,r31
-	sbc r9,r31
 	mov r2,r24
 	mov __tmp_reg__,r24
 	lsl r0
 	sbc r3,r3
 	cpse r24,__zero_reg__
 	rjmp .L32
-	ldi r30,lo8(1)
+	ldi r31,lo8(1)
 	adiw r28,82-63
-	std Y+63,r30
+	std Y+63,r31
 	sbiw r28,82-63
 	mov r2,__zero_reg__
 	mov r3,__zero_reg__
@@ -936,10 +939,10 @@
 	mov r4,__zero_reg__
 	ldi r30,lo8(50)
 	ldi r31,0
-	ldi r24,lo8(.LC1+2)
-	mov r8,r24
-	ldi r24,hi8(.LC1+2)
-	mov r9,r24
+	ldi r25,lo8(.LC1+2)
+	mov r8,r25
+	ldi r25,hi8(.LC1+2)
+	mov r9,r25
 .L33:
 	adiw r28,82-63
 	ldd r18,Y+63
@@ -1017,21 +1020,23 @@
 	std Y+63,r24
 	sbiw r28,86-63
 	mov r4,r25
-	movw r30,r8
+	movw r24,r8
+	ldi r18,-1
+	sub r8,r18
+	sbc r9,r18
+	movw r30,r24
 	ld r24,Z
-	ldi r31,-1
-	sub r8,r31
-	sbc r9,r31
 	mov r30,r24
 	mov __tmp_reg__,r24
 	lsl r0
 	sbc r31,r31
 	cpse r24,__zero_reg__
 	rjmp .L33
-	movw r18,r28
-	subi r18,-1
-	sbci r19,-1
-	movw r8,r18
+	clr r8
+	inc r8
+	mov r9,__zero_reg__
+	add r8,r28
+	adc r9,r29
 	adiw r28,81-63
 	ldd r10,Y+63
 	sbiw r28,81-63
