DIR cf0ee995841b1b353959cc06283c1400
--- pr81290.c.{avr25}.{3}.{vanilla}.s	2020-08-13 19:00:29.140155570 +0000
+++ pr81290.c.{avr25}.{3}.{ccmode}.s	2020-08-13 19:00:30.520132225 +0000
@@ -16,6 +16,7 @@
 	ldi r24,lo8(19)
 	ldi r25,0
 .L2:
+	movw r20,r18
 	lsl r18
 	rol r19
 	sbiw r24,1
@@ -24,7 +25,7 @@
 	ldi r25,0
 	sts vz+1,r25
 	sts vz,r24
-	or r18,r19
+	or r20,r21
 	brne .L4
 	rcall abort
 .L4:
