DIR c8a6abd1aaf75fce355a63c05060dc59
--- pr51323.c.{avr25}.{s}.{vanilla}.s	2020-08-13 18:39:37.172125321 +0000
+++ pr51323.c.{avr25}.{s}.{ccmode}.s	2020-08-13 18:39:38.604112493 +0000
@@ -75,32 +75,24 @@
 /* frame size = 6 */
 /* stack size = 6 */
 .L__stack_usage = 6
-	ldi r24,lo8(3)
-	ldi r25,0
-	sts v,r24
-	sts v+1,r25
-	ldi r19,lo8(2)
-	mov r18,r19
-	mov r20,r24
-	ldi r24,lo8(4)
-	mov r22,r24
+	ldi r20,lo8(3)
+	sts v,r20
+	sts v+1,__zero_reg__
+	ldi r18,lo8(2)
 	ldi r19,0
 	ldi r21,0
+	ldi r22,lo8(4)
 	ldi r23,0
 	ldi r24,lo8(9)
 	ldi r25,0
 	rcall bar
-	ldi r24,lo8(17)
-	ldi r25,0
-	sts v,r24
-	sts v+1,r25
-	ldi r19,lo8(16)
-	mov r18,r19
-	mov r20,r24
-	ldi r24,lo8(18)
-	mov r22,r24
+	ldi r20,lo8(17)
+	sts v,r20
+	sts v+1,__zero_reg__
+	ldi r18,lo8(16)
 	ldi r19,0
 	ldi r21,0
+	ldi r22,lo8(18)
 	ldi r23,0
 	ldi r24,lo8(9)
 	ldi r25,0
