--- 20020413-1.c.{atmega128}.{s}.{vanilla}.s	2020-08-05 15:45:57.736737967 +0000
+++ 20020413-1.c.{atmega128}.{s}.{ccmode}.s	2020-08-05 15:45:57.520735803 +0000
@@ -278,6 +278,9 @@
 	ldi r24,lo8(8)
 	ldi r25,lo8(64)
 	call test
+	movw r16,r28
+	subi r16,-1
+	sbci r17,-1
 	ldi r18,0
 	ldi r19,0
 	ldi r20,0
@@ -287,6 +290,9 @@
 	ldi r24,lo8(12)
 	ldi r25,lo8(64)
 	call test
+	movw r16,r28
+	subi r16,-1
+	sbci r17,-1
 	ldi r18,0
 	ldi r19,0
 	ldi r20,0
@@ -296,6 +302,9 @@
 	ldi r24,lo8(16)
 	ldi r25,lo8(64)
 	call test
+	movw r16,r28
+	subi r16,-1
+	sbci r17,-1
 	ldi r18,0
 	ldi r19,0
 	ldi r20,0
