--- pr64979.c.{atmega128}.{3}.{vanilla}.s	2020-08-05 15:53:36.969331987 +0000
+++ pr64979.c.{atmega128}.{3}.{ccmode}.s	2020-08-05 15:53:36.713329429 +0000
@@ -33,43 +33,52 @@
 	ldd r24,Z+2
 	ldd r25,Z+3
 	sbiw r24,1
-	brne .L3
-	adiw r26,2
+	breq .+2
+	rjmp .L3
+	movw r26,r30
+	adiw r26,6
 	ldd r24,Z+4
 	ldd r25,Z+5
 	sbiw r24,2
 	brne .L3
-	adiw r26,2
+	movw r26,r30
+	adiw r26,8
 	ldd r24,Z+6
 	ldd r25,Z+7
 	sbiw r24,3
 	brne .L3
-	adiw r26,2
+	movw r26,r30
+	adiw r26,10
 	ldd r24,Z+8
 	ldd r25,Z+9
 	sbiw r24,4
 	brne .L3
-	adiw r26,2
+	movw r26,r30
+	adiw r26,12
 	ldd r24,Z+10
 	ldd r25,Z+11
 	sbiw r24,5
 	brne .L3
-	adiw r26,2
+	movw r26,r30
+	adiw r26,14
 	ldd r24,Z+12
 	ldd r25,Z+13
 	sbiw r24,6
 	brne .L3
-	adiw r26,2
+	movw r26,r30
+	adiw r26,16
 	ldd r24,Z+14
 	ldd r25,Z+15
 	sbiw r24,7
 	brne .L3
-	adiw r26,2
+	movw r26,r30
+	adiw r26,18
 	ldd r24,Z+16
 	ldd r25,Z+17
 	sbiw r24,8
 	brne .L3
-	adiw r26,2
+	movw r26,r30
+	adiw r26,20
 	ldd r24,Z+18
 	ldd r25,Z+19
 	sbiw r24,9
