DIR 1e99d5f9bc757db22bef2868e50ef8a0
--- 20000113-1.c.{atmega128}.{s}.{vanilla}.s	2020-08-05 18:45:16.524828893 +0000
+++ 20000113-1.c.{atmega128}.{s}.{ccmode}.s	2020-08-05 18:45:16.368827832 +0000
@@ -15,20 +15,21 @@
 .L__stack_usage = 0
 	movw r18,r24
 	movw r24,r22
-	mov r21,r22
-	andi r21,lo8(3)
+	mov r21,r18
+	andi r21,lo8(1)
+	andi r22,lo8(3)
 	andi r24,3
 	clr r25
-	movw r22,r18
-	andi r22,1
-	clr r23
-	sub r24,r22
-	sbc r25,r23
-	mul r21,r24
+	andi r18,1
+	clr r19
+	sub r24,r18
+	sbc r25,r19
+	mul r22,r24
 	add r20,r0
 	clr __zero_reg__
 	andi r20,lo8(7)
-	sbrs r18,0
+	cpse r21,__zero_reg__
+	rjmp .L2
 .L3:
 	call abort
 .L2:
