--- pr51877.c.{atmega128}.{3}.{vanilla}.s	2020-08-05 15:53:25.141213803 +0000
+++ pr51877.c.{atmega128}.{3}.{ccmode}.s	2020-08-05 15:53:24.769210086 +0000
@@ -41,7 +41,8 @@
 	brne 0b
 	std Y+3,r22
 	ldi r18,lo8(34)
-	sbiw r30,3
+	movw r30,r28
+	adiw r30,1
 	movw r26,r24
 	0:
 	ld r0,Z+
@@ -91,11 +92,13 @@
 	movw r16,r24
 	cpi r22,6
 	cpc r23,__zero_reg__
-	breq .L7
 	ldi r22,lo8(7)
 	ldi r23,0
-	movw r24,r28
-	adiw r24,1
+	ldi r24,lo8(1)
+	ldi r25,0
+	breq .L7
+	add r24,r28
+	adc r25,r29
 	call bar
 	ldi r24,lo8(34)
 	movw r30,r28
@@ -120,10 +123,8 @@
 	pop r16
 	jmp baz
 .L7:
-	ldi r22,lo8(7)
-	ldi r23,0
-	movw r24,r28
-	adiw r24,1
+	add r24,r28
+	adc r25,r29
 	call bar
 	ldi r24,lo8(34)
 	movw r30,r28
