--- pr68532.c.{atmega128}.{3}.{vanilla}.s	2020-08-05 15:52:55.048913102 +0000
+++ pr68532.c.{atmega128}.{3}.{ccmode}.s	2020-08-05 15:52:54.984912463 +0000
@@ -33,64 +33,74 @@
 	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,-80
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,-96
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,-112
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,-128
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,112
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,96
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,80
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,64
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,48
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
-	sbiw r26,1
 	add r18,r22
 	adc r19,r23
-	adiw r26,16
+	movw r26,r30
+	subi r26,32
+	sbci r27,-1
 	ld r22,X+
 	ld r23,X
 	add r18,r22
