DIR 7d1e785a0a9abfb6db53f15245ad00d1
--- pr60655-2.c.{atmega128}.{s}.{vanilla}.s	2020-08-05 19:03:09.741464389 +0000
+++ pr60655-2.c.{atmega128}.{s}.{lra}.s	2020-08-06 13:13:15.837338843 +0000
@@ -15,12 +15,12 @@
 .L__stack_usage = 0
 	lds r18,global_precision
 	lds r19,global_precision+1
+	add r24,r18
+	adc r25,r19
 	movw r30,r24
-	add r30,r18
-	adc r31,r19
+	add r22,r18
+	adc r23,r19
 	movw r26,r22
-	add r26,r18
-	adc r27,r19
 .L3:
 	ld r25,-Z
 	ld r24,-X
@@ -45,13 +45,14 @@
 .global	upton_modmult
 	.type	upton_modmult, @function
 upton_modmult:
-	push r28
+	push r17
 	push r29
 /* prologue: function */
 /* frame size = 0 */
 /* stack size = 2 */
 .L__stack_usage = 2
-	movw r28,r24
+	mov r29,r24
+	mov r17,r25
 .L8:
 	ldi r22,lo8(modulus)
 	ldi r23,hi8(modulus)
@@ -63,14 +64,15 @@
 	brlt .L9
 /* epilogue start */
 	pop r29
-	pop r28
+	pop r17
 	ret
 .L9:
 	lds r20,global_precision
 	lds r21,global_precision+1
 	ldi r22,lo8(d_data)
 	ldi r23,hi8(d_data)
-	movw r24,r28
+	mov r24,r29
+	mov r25,r17
 	call memcpy
 	rjmp .L8
 	.size	upton_modmult, .-upton_modmult
