DIR 45c2df963dbc9404d6591934dce02c64
--- pr27260.c.{atmega128}.{s}.{vanilla}.s	2020-08-05 19:00:03.591642362 +0000
+++ pr27260.c.{atmega128}.{s}.{lra}.s	2020-08-06 13:10:35.527860813 +0000
@@ -36,9 +36,9 @@
 .L__stack_usage = 0
 	ldi r24,lo8(2)
 	sts buf+64,r24
-	ldi r28,lo8(buf)
-	ldi r29,hi8(buf)
-	movw r30,r28
+	ldi r16,lo8(buf)
+	ldi r17,hi8(buf)
+	movw r30,r16
 .L6:
 	ld r24,Z+
 	cpse r24,__zero_reg__
@@ -66,12 +66,13 @@
 	ldi r25,0
 	call foo
 .L9:
-	ld r24,Y+
+	movw r30,r16
+	ld r24,Z+
+	movw r16,r30
 	cpse r24,__zero_reg__
 	rjmp .L7
-	ldi r24,hi8(buf+64)
-	cpi r28,lo8(buf+64)
-	cpc r29,r24
+	subi r30,lo8(buf+64)
+	sbci r31,hi8(buf+64)
 	brne .L9
 	lds r24,buf+64
 	cpi r24,lo8(2)
