DIR 8d42ad42d78d61787603046333d99a46
--- dbra-1.c.{avr3}.{3}.{vanilla}.s	2020-08-10 18:54:37.679757500 +0000
+++ dbra-1.c.{avr3}.{3}.{ccmode}.s	2020-08-10 18:54:43.383758979 +0000
@@ -171,83 +171,61 @@
 	mov r21,r23
 	mov r22,r24
 	mov r23,r25
-	mov r27,r25
-	mov r26,r24
-	mov r25,r18
-	mov r24,r19
-	sbiw r24,1
-	sbc r26,__zero_reg__
-	sbc r27,__zero_reg__
-	brne .+2
-	rjmp .L24
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	sbiw r24,2
-	sbc r26,__zero_reg__
-	sbc r27,__zero_reg__
-	brne .+2
-	rjmp .L25
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	sbiw r24,3
-	sbc r26,__zero_reg__
-	sbc r27,__zero_reg__
-	brne .+2
-	rjmp .L26
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	sbiw r24,4
-	sbc r26,__zero_reg__
-	sbc r27,__zero_reg__
-	brne .+2
-	rjmp .L27
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	sbiw r24,5
-	sbc r26,__zero_reg__
-	sbc r27,__zero_reg__
+	cpi r20,1
+	cpc r21,__zero_reg__
+	cpc r22,__zero_reg__
+	cpc r23,__zero_reg__
+	breq .L24
+	cpi r20,2
+	cpc r21,__zero_reg__
+	cpc r22,__zero_reg__
+	cpc r23,__zero_reg__
+	breq .L25
+	cpi r20,3
+	cpc r21,__zero_reg__
+	cpc r22,__zero_reg__
+	cpc r23,__zero_reg__
+	breq .L26
+	cpi r20,4
+	cpc r21,__zero_reg__
+	cpc r22,__zero_reg__
+	cpc r23,__zero_reg__
+	breq .L27
+	cpi r20,5
+	cpc r21,__zero_reg__
+	cpc r22,__zero_reg__
+	cpc r23,__zero_reg__
 	breq .L28
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	sbiw r24,6
-	sbc r26,__zero_reg__
-	sbc r27,__zero_reg__
+	cpi r20,6
+	cpc r21,__zero_reg__
+	cpc r22,__zero_reg__
+	cpc r23,__zero_reg__
 	breq .L29
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	sbiw r24,7
-	sbc r26,__zero_reg__
-	sbc r27,__zero_reg__
+	cpi r20,7
+	cpc r21,__zero_reg__
+	cpc r22,__zero_reg__
+	cpc r23,__zero_reg__
 	breq .L30
-	mov r27,r23
-	mov r26,r22
+	cpi r20,8
+	cpc r21,__zero_reg__
+	cpc r22,__zero_reg__
+	cpc r23,__zero_reg__
+	breq .L31
+	mov r27,r25
+	mov r26,r24
 	mov r25,r18
 	mov r24,r19
-	sbiw r24,8
+	sbiw r24,9
 	sbc r26,__zero_reg__
 	sbc r27,__zero_reg__
-	breq .L31
-	subi r20,9
-	sbc r21,__zero_reg__
-	sbc r22,__zero_reg__
-	sbc r23,__zero_reg__
-	breq .L32
-	cpi r20,1
+	cpi r20,9
 	cpc r21,__zero_reg__
 	cpc r22,__zero_reg__
 	cpc r23,__zero_reg__
+	breq .L32
+	sbiw r24,1
+	cpc r26,__zero_reg__
+	cpc r27,__zero_reg__
 	breq .L34
 	ldi r24,lo8(-1)
 	ldi r25,lo8(-1)
@@ -342,112 +320,74 @@
 /* stack size = 0 */
 .L__stack_usage = 0
 	mov r19,r22
-	mov r18,r23
 	mov r20,r22
 	mov r21,r23
 	mov r22,r24
 	mov r23,r25
-	mov r27,r25
-	mov r26,r24
-	mov r25,r18
-	mov r24,r19
-	adiw r24,1
-	adc r26,__zero_reg__
-	adc r27,__zero_reg__
-	or r24,r25
-	or r24,r26
-	or r24,r27
-	brne .+2
-	rjmp .L43
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	adiw r24,2
-	adc r26,__zero_reg__
-	adc r27,__zero_reg__
-	or r24,r25
-	or r24,r26
-	or r24,r27
-	brne .+2
-	rjmp .L44
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	adiw r24,3
-	adc r26,__zero_reg__
-	adc r27,__zero_reg__
-	or r24,r25
-	or r24,r26
-	or r24,r27
-	brne .+2
-	rjmp .L45
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	adiw r24,4
-	adc r26,__zero_reg__
-	adc r27,__zero_reg__
-	or r24,r25
-	or r24,r26
-	or r24,r27
-	brne .+2
-	rjmp .L46
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	adiw r24,5
-	adc r26,__zero_reg__
-	adc r27,__zero_reg__
-	or r24,r25
-	or r24,r26
-	or r24,r27
+	cpi r20,-1
+	ldi r18,-1
+	cpc r21,r18
+	cpc r22,r18
+	cpc r23,r18
+	breq .L43
+	cpi r20,-2
+	ldi r24,-1
+	cpc r21,r24
+	cpc r22,r24
+	cpc r23,r24
+	breq .L44
+	cpi r20,-3
+	ldi r18,-1
+	cpc r21,r18
+	cpc r22,r18
+	cpc r23,r18
+	breq .L45
+	cpi r20,-4
+	ldi r24,-1
+	cpc r21,r24
+	cpc r22,r24
+	cpc r23,r24
+	breq .L46
+	cpi r20,-5
+	ldi r18,-1
+	cpc r21,r18
+	cpc r22,r18
+	cpc r23,r18
 	breq .L47
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	adiw r24,6
-	adc r26,__zero_reg__
-	adc r27,__zero_reg__
-	or r24,r25
-	or r24,r26
-	or r24,r27
+	cpi r20,-6
+	ldi r24,-1
+	cpc r21,r24
+	cpc r22,r24
+	cpc r23,r24
 	breq .L48
-	mov r27,r23
-	mov r26,r22
-	mov r25,r18
-	mov r24,r19
-	adiw r24,7
-	adc r26,__zero_reg__
-	adc r27,__zero_reg__
-	or r24,r25
-	or r24,r26
-	or r24,r27
+	cpi r20,-7
+	ldi r18,-1
+	cpc r21,r18
+	cpc r22,r18
+	cpc r23,r18
 	breq .L49
-	mov r27,r23
+	cpi r20,-8
+	ldi r24,-1
+	cpc r21,r24
+	cpc r22,r24
+	cpc r23,r24
+	breq .L50
+	mov r27,r25
 	mov r26,r22
-	mov r25,r18
+	mov r25,r21
 	mov r24,r19
-	adiw r24,8
+	adiw r24,9
 	adc r26,__zero_reg__
 	adc r27,__zero_reg__
-	or r24,r25
-	or r24,r26
-	or r24,r27
-	breq .L50
-	subi r20,-9
+	cpi r20,-9
 	sbci r21,-1
 	sbci r22,-1
 	sbci r23,-1
 	breq .L51
-	cpi r20,-1
-	sbci r21,-1
-	sbci r22,-1
-	sbci r23,-1
+	cpi r24,-1
+	sbci r25,-1
+	sbci r26,-1
+	sbci r27,-1
 	breq .L53
 	ldi r24,lo8(-1)
 	ldi r25,lo8(-1)
