DIR 0ac88e7c933333769fdc0c16e81d6349
--- pr77754-6.c.{atmega128}.{s}.{vanilla}.s	2020-08-14 12:24:48.374785710 +0000
+++ pr77754-6.c.{atmega128}.{s}.{ccmode}.s	2020-08-14 12:24:50.570753074 +0000
@@ -31,14 +31,14 @@
 	call fn3
 	movw r16,r24
 	call fn3
-	ldi r19,2
-	add r14,r19
-	adc r15,__zero_reg__
-	mul r14,r24
+	movw r18,r14
+	subi r18,-2
+	sbci r19,-1
+	mul r18,r24
 	movw r20,r0
-	mul r14,r25
+	mul r18,r25
 	add r21,r0
-	mul r15,r24
+	mul r19,r24
 	add r21,r0
 	clr r1
 	subi r16,-1
