--- 20131127-1.c.{atmega128}.{3}.{vanilla}.s	2020-08-05 14:14:49.737216500 +0000
+++ 20131127-1.c.{atmega128}.{3}.{ccmode}.s	2020-08-05 14:14:49.557214310 +0000
@@ -26,8 +26,10 @@
 	ldi r24,lo8(8)
 	ldi r30,lo8(c)
 	ldi r31,hi8(c)
-	movw r26,r28
-	adiw r26,1
+	ldi r26,lo8(1)
+	ldi r27,0
+	add r26,r28
+	adc r27,r29
 	0:
 	ld r0,Z+
 	st X+,r0
@@ -72,16 +74,20 @@
 	ldi r24,lo8(8)
 	ldi r30,lo8(c)
 	ldi r31,hi8(c)
-	movw r26,r28
-	adiw r26,1
+	ldi r26,lo8(1)
+	ldi r27,0
+	add r26,r28
+	adc r27,r29
 	0:
 	ld r0,Z+
 	st X+,r0
 	dec r24
 	brne 0b
 	ldi r24,lo8(8)
-	movw r30,r28
-	adiw r30,1
+	ldi r30,lo8(1)
+	ldi r31,0
+	add r30,r28
+	adc r31,r29
 	movw r26,r28
 	adiw r26,9
 	0:
