DIR 93bf0cd0d02d219804f0d2e10c1385d7
--- vfprintf-1.c.{atmega128}.{s}.{vanilla}.s	2020-08-07 14:42:15.271990364 +0000
+++ vfprintf-1.c.{atmega128}.{s}.{ccmode}.s	2020-08-07 14:42:15.939995929 +0000
@@ -44,8 +44,14 @@
 	ld r30,X+
 	ld r31,X+
 	movw r16,r26
-	cpi r30,11
-	cpc r31,__zero_reg__
+	movw r24,r30
+	mov __tmp_reg__,r31
+	lsl r0
+	sbc r26,r26
+	sbc r27,r27
+	sbiw r24,11
+	cpc r26,__zero_reg__
+	cpc r27,__zero_reg__
 	brsh .L2
 	subi r30,lo8(-(gs(.L4)))
 	sbci r31,hi8(-(gs(.L4)))
@@ -75,12 +81,14 @@
 	movw r20,r16
 	ldi r22,lo8(.LC0)
 	ldi r23,hi8(.LC0)
-.L27:
+.L28:
 	lds r24,__iob+2
 	lds r25,__iob+2+1
 	call vfprintf
 	sbiw r24,5
-	breq .L1
+.L21:
+	brne .+2
+	rjmp .L1
 .L2:
 	call abort
 .L13:
@@ -91,19 +99,12 @@
 	movw r20,r16
 	ldi r22,lo8(.LC1)
 	ldi r23,hi8(.LC1)
-.L21:
+.L32:
 	lds r24,__iob+2
 	lds r25,__iob+2+1
 	call vfprintf
 	sbiw r24,6
-	brne .L2
-.L1:
-/* epilogue start */
-	pop r29
-	pop r28
-	pop r17
-	pop r16
-	ret
+	rjmp .L21
 .L12:
 	movw r22,r24
 	ldi r24,lo8(97)
@@ -112,51 +113,49 @@
 	movw r20,r16
 	ldi r22,lo8(.LC2)
 	ldi r23,hi8(.LC2)
-.L23:
+.L33:
 	lds r24,__iob+2
 	lds r25,__iob+2+1
 	call vfprintf
 	sbiw r24,1
-	breq .L1
-	rjmp .L2
+	rjmp .L21
 .L11:
-	movw r20,r26
+	movw r20,r16
 	ldi r22,lo8(.LC3)
 	ldi r23,hi8(.LC3)
-.L25:
+.L34:
 	call vfprintf
-	or r24,r25
-	breq .L1
-	rjmp .L2
+	sbiw r24,0
+	rjmp .L21
 .L10:
-	movw r20,r26
+	movw r20,r16
 	ldi r22,lo8(.LC4)
 	ldi r23,hi8(.LC4)
 	call vfprintf
 	movw r20,r16
 	ldi r22,lo8(.LC4)
 	ldi r23,hi8(.LC4)
-	rjmp .L27
+	rjmp .L28
 .L9:
-	movw r20,r26
+	movw r20,r16
 	ldi r22,lo8(.LC4)
 	ldi r23,hi8(.LC4)
 	call vfprintf
 	movw r20,r16
 	ldi r22,lo8(.LC4)
 	ldi r23,hi8(.LC4)
-	rjmp .L21
+	rjmp .L32
 .L8:
-	movw r20,r26
+	movw r20,r16
 	ldi r22,lo8(.LC4)
 	ldi r23,hi8(.LC4)
 	call vfprintf
 	movw r20,r16
 	ldi r22,lo8(.LC4)
 	ldi r23,hi8(.LC4)
-	rjmp .L23
+	rjmp .L33
 .L7:
-	movw r20,r26
+	movw r20,r16
 	ldi r22,lo8(.LC4)
 	ldi r23,hi8(.LC4)
 	call vfprintf
@@ -165,18 +164,18 @@
 	ldi r23,hi8(.LC4)
 	lds r24,__iob+2
 	lds r25,__iob+2+1
-	rjmp .L25
+	rjmp .L34
 .L6:
-	movw r20,r26
+	movw r20,r16
 	ldi r22,lo8(.LC5)
 	ldi r23,hi8(.LC5)
 	call vfprintf
 	movw r20,r16
 	ldi r22,lo8(.LC5)
 	ldi r23,hi8(.LC5)
-	rjmp .L23
+	rjmp .L33
 .L5:
-	movw r20,r26
+	movw r20,r16
 	ldi r22,lo8(.LC6)
 	ldi r23,hi8(.LC6)
 	call vfprintf
@@ -187,11 +186,9 @@
 	lds r25,__iob+2+1
 	call vfprintf
 	sbiw r24,7
-	brne .+2
-	rjmp .L1
-	rjmp .L2
+	rjmp .L21
 .L3:
-	movw r20,r26
+	movw r20,r16
 	ldi r22,lo8(.LC7)
 	ldi r23,hi8(.LC7)
 	call vfprintf
@@ -202,9 +199,14 @@
 	lds r25,__iob+2+1
 	call vfprintf
 	sbiw r24,2
-	brne .+2
-	rjmp .L1
-	rjmp .L2
+	rjmp .L21
+.L1:
+/* epilogue start */
+	pop r29
+	pop r28
+	pop r17
+	pop r16
+	ret
 	.size	inner, .-inner
 	.section	.text.startup,"ax",@progbits
 .global	main
