DIR 6af91fa92406df2d4052fb5e06f7fa79
--- simd-2.c.{atmega128}.{3}.{vanilla}.s	2020-08-05 18:49:44.160164240 +0000
+++ simd-2.c.{atmega128}.{3}.{ccmode}.s	2020-08-05 18:49:42.812144033 +0000
@@ -70,159 +70,163 @@
 /* frame size = 72 */
 /* stack size = 72 */
 .L__stack_usage = 72
-	lds r15,i
-	std Y+11,r15
-	lds r16,i+1
-	std Y+12,r16
-	lds r17,i+2
-	std Y+13,r17
-	lds r18,i+3
-	std Y+14,r18
-	lds r20,i+4
-	std Y+15,r20
-	lds r30,i+5
-	std Y+16,r30
-	lds r31,i+6
-	std Y+17,r31
-	lds r15,i+7
-	std Y+18,r15
-	lds r16,i+8
-	std Y+19,r16
-	lds r17,i+9
-	std Y+20,r17
-	lds r18,i+10
-	std Y+21,r18
-	lds r20,i+11
-	std Y+22,r20
-	lds r30,i+12
-	std Y+23,r30
-	lds r31,i+13
-	std Y+24,r31
-	lds r15,i+14
-	std Y+25,r15
-	lds r16,i+15
-	std Y+26,r16
-	lds r17,j
-	std Y+35,r17
-	lds r18,j+1
-	std Y+36,r18
-	lds r20,j+2
-	std Y+37,r20
-	lds r30,j+3
-	std Y+38,r30
-	lds r31,j+4
-	std Y+39,r31
-	lds r15,j+5
-	std Y+40,r15
-	lds r16,j+6
-	std Y+41,r16
-	lds r17,j+7
-	std Y+42,r17
-	lds r18,j+8
-	std Y+43,r18
-	lds r20,j+9
-	std Y+44,r20
-	lds r30,j+10
-	std Y+45,r30
-	lds r31,j+11
-	std Y+46,r31
-	lds r15,j+12
-	std Y+47,r15
-	lds r16,j+13
-	std Y+48,r16
-	lds r17,j+14
-	std Y+49,r17
-	lds r18,j+15
-	std Y+50,r18
-	ldd r6,Y+11
-	ldd r7,Y+12
-	ldd r20,Y+35
-	std Y+1,r20
-	ldd r30,Y+36
-	std Y+2,r30
-	ldd r30,Y+1
-	ldd r31,Y+2
-	add r30,r6
-	adc r31,r7
-	ldd r8,Y+13
-	ldd r9,Y+14
-	ldd r15,Y+37
-	std Y+27,r15
-	ldd r16,Y+38
-	std Y+28,r16
-	ldd r22,Y+27
-	ldd r23,Y+28
-	add r22,r8
-	adc r23,r9
-	ldd r10,Y+15
-	ldd r11,Y+16
+	lds r14,i
+	std Y+11,r14
+	lds r15,i+1
+	std Y+12,r15
+	lds r16,i+2
+	std Y+13,r16
+	lds r17,i+3
+	std Y+14,r17
+	lds r18,i+4
+	std Y+15,r18
+	lds r20,i+5
+	std Y+16,r20
+	lds r30,i+6
+	std Y+17,r30
+	lds r14,i+7
+	std Y+18,r14
+	lds r15,i+8
+	std Y+19,r15
+	lds r16,i+9
+	std Y+20,r16
+	lds r17,i+10
+	std Y+21,r17
+	lds r18,i+11
+	std Y+22,r18
+	lds r20,i+12
+	std Y+23,r20
+	lds r30,i+13
+	std Y+24,r30
+	lds r14,i+14
+	std Y+25,r14
+	lds r15,i+15
+	std Y+26,r15
+	lds r16,j
+	std Y+35,r16
+	lds r17,j+1
+	std Y+36,r17
+	lds r18,j+2
+	std Y+37,r18
+	lds r20,j+3
+	std Y+38,r20
+	lds r30,j+4
+	std Y+39,r30
+	lds r14,j+5
+	std Y+40,r14
+	lds r15,j+6
+	std Y+41,r15
+	lds r16,j+7
+	std Y+42,r16
+	lds r17,j+8
+	std Y+43,r17
+	lds r18,j+9
+	std Y+44,r18
+	lds r20,j+10
+	std Y+45,r20
+	lds r30,j+11
+	std Y+46,r30
+	lds r14,j+12
+	std Y+47,r14
+	lds r15,j+13
+	std Y+48,r15
+	lds r16,j+14
+	std Y+49,r16
+	lds r17,j+15
+	std Y+50,r17
+	ldd r18,Y+11
+	std Y+1,r18
+	ldd r20,Y+12
+	std Y+2,r20
+	ldd r30,Y+35
+	std Y+27,r30
+	ldd r14,Y+36
+	std Y+28,r14
+	ldd r30,Y+27
+	ldd r31,Y+28
+	ldd r16,Y+1
+	ldd r17,Y+2
+	add r30,r16
+	adc r31,r17
+	ldd r6,Y+13
+	ldd r7,Y+14
+	ldd r17,Y+37
+	std Y+29,r17
+	ldd r18,Y+38
+	std Y+30,r18
+	ldd r22,Y+29
+	ldd r23,Y+30
+	add r22,r6
+	adc r23,r7
+	ldd r8,Y+15
+	ldd r9,Y+16
 	ldd r2,Y+39
 	ldd r3,Y+40
-	movw r20,r10
+	movw r20,r8
 	add r20,r2
 	adc r21,r3
-	ldd r12,Y+17
-	ldd r13,Y+18
-	ldd r17,Y+41
-	std Y+29,r17
-	ldd r18,Y+42
-	std Y+30,r18
-	ldd r24,Y+29
-	ldd r25,Y+30
-	add r24,r12
-	adc r25,r13
-	ldd r15,Y+19
-	std Y+3,r15
-	ldd r16,Y+20
-	std Y+4,r16
+	ldd r10,Y+17
+	ldd r11,Y+18
+	ldd r14,Y+41
+	std Y+31,r14
+	ldd r15,Y+42
+	std Y+32,r15
+	ldd r24,Y+31
+	ldd r25,Y+32
+	add r24,r10
+	adc r25,r11
+	ldd r16,Y+19
+	std Y+3,r16
+	ldd r17,Y+20
+	std Y+4,r17
 	ldd r4,Y+43
 	ldd r5,Y+44
-	ldd r17,Y+21
-	std Y+5,r17
-	ldd r18,Y+22
-	std Y+6,r18
+	ldd r18,Y+21
+	std Y+5,r18
+	ldd r14,Y+22
+	std Y+6,r14
 	ldd r15,Y+45
-	std Y+31,r15
+	std Y+33,r15
 	ldd r16,Y+46
-	std Y+32,r16
+	std Y+34,r16
 	ldd r17,Y+23
 	std Y+7,r17
 	ldd r18,Y+24
 	std Y+8,r18
-	ldd r15,Y+47
-	std Y+33,r15
-	ldd r16,Y+48
-	std Y+34,r16
-	ldd r17,Y+25
-	std Y+9,r17
-	ldd r18,Y+26
-	std Y+10,r18
-	ldd r18,Y+49
-	ldd r19,Y+50
-	ldd r14,Y+3
-	ldd r15,Y+4
-	add r14,r4
-	adc r15,r5
-	ldd r16,Y+5
-	ldd r17,Y+6
-	ldd r26,Y+31
-	ldd r27,Y+32
-	add r16,r26
-	adc r17,r27
-	std Y+54,r17
-	std Y+53,r16
+	ldd r18,Y+47
+	ldd r19,Y+48
+	ldd r14,Y+25
+	std Y+9,r14
+	ldd r15,Y+26
+	std Y+10,r15
+	ldd r16,Y+49
+	std Y+51,r16
+	ldd r17,Y+50
+	std Y+52,r17
+	ldd r12,Y+3
+	ldd r13,Y+4
+	add r12,r4
+	adc r13,r5
+	ldd r14,Y+5
+	ldd r15,Y+6
+	ldd r16,Y+33
+	ldd r17,Y+34
+	add r14,r16
+	adc r15,r17
 	ldd r16,Y+7
 	ldd r17,Y+8
-	ldd r26,Y+33
-	ldd r27,Y+34
-	add r16,r26
-	adc r17,r27
-	std Y+52,r17
-	std Y+51,r16
+	add r16,r18
+	adc r17,r19
+	adiw r28,67-62
+	std Y+63,r17
+	std Y+62,r16
+	sbiw r28,67-62
 	ldd r26,Y+9
 	ldd r27,Y+10
-	add r26,r18
-	adc r27,r19
+	ldd r16,Y+51
+	ldd r17,Y+52
+	add r26,r16
+	adc r27,r17
 	sts k,r30
 	sts k+1,r31
 	sts k+2,r22
@@ -231,16 +235,18 @@
 	sts k+5,r21
 	sts k+6,r24
 	sts k+7,r25
-	sts k+8,r14
-	sts k+9,r15
-	ldd r16,Y+53
-	sts k+10,r16
-	ldd r17,Y+54
-	sts k+11,r17
-	ldd r16,Y+51
-	sts k+12,r16
-	ldd r17,Y+52
-	sts k+13,r17
+	sts k+8,r12
+	sts k+9,r13
+	sts k+10,r14
+	sts k+11,r15
+	adiw r28,67-63
+	ldd r17,Y+63
+	sbiw r28,67-63
+	sts k+12,r17
+	adiw r28,68-63
+	ldd r16,Y+63
+	sbiw r28,68-63
+	sts k+13,r16
 	sts k+14,r26
 	sts k+15,r27
 	sts res,r30
@@ -251,16 +257,12 @@
 	sts res+5,r21
 	sts res+6,r24
 	sts res+7,r25
-	sts res+8,r14
-	sts res+9,r15
-	ldd r15,Y+53
-	sts res+10,r15
-	ldd r16,Y+54
-	sts res+11,r16
-	ldd r17,Y+51
+	sts res+8,r12
+	sts res+9,r13
+	sts res+10,r14
+	sts res+11,r15
 	sts res+12,r17
-	ldd r15,Y+52
-	sts res+13,r15
+	sts res+13,r16
 	sts res+14,r26
 	sts res+15,r27
 	cpi r30,-96
@@ -279,176 +281,178 @@
 	cpc r25,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldd r16,Y+1
-	ldd r17,Y+2
-	mul r16,r6
+	ldd r16,Y+27
+	ldd r17,Y+28
+	ldd r20,Y+1
+	ldd r21,Y+2
+	mul r16,r20
+	movw r30,r0
+	mul r16,r21
+	add r31,r0
+	mul r17,r20
+	add r31,r0
+	clr r1
+	ldd r14,Y+29
+	ldd r15,Y+30
+	mul r14,r6
+	movw r22,r0
+	mul r14,r7
+	add r23,r0
+	mul r15,r6
+	add r23,r0
+	clr r1
+	mul r8,r2
+	movw r20,r0
+	mul r8,r3
+	add r21,r0
+	mul r9,r2
+	add r21,r0
+	clr r1
+	ldd r16,Y+31
+	ldd r17,Y+32
+	mul r10,r16
 	movw r24,r0
-	mul r16,r7
+	mul r10,r17
 	add r25,r0
-	mul r17,r6
+	mul r11,r16
 	add r25,r0
 	clr r1
-	ldd r20,Y+27
-	ldd r21,Y+28
-	mul r20,r8
-	movw r14,r0
-	mul r20,r9
-	add r15,r0
-	mul r21,r8
-	add r15,r0
+	ldd r14,Y+3
+	ldd r15,Y+4
+	mul r14,r4
+	movw r12,r0
+	mul r14,r5
+	add r13,r0
+	mul r15,r4
+	add r13,r0
 	clr r1
-	mul r10,r2
+	ldd r14,Y+5
+	ldd r15,Y+6
+	ldd r26,Y+33
+	ldd r27,Y+34
+	mul r14,r26
 	movw r16,r0
-	mul r10,r3
+	mul r14,r27
 	add r17,r0
-	mul r11,r2
+	mul r15,r26
 	add r17,r0
 	clr r1
-	ldd r30,Y+29
-	ldd r31,Y+30
-	mul r12,r30
-	movw r22,r0
-	mul r12,r31
-	add r23,r0
-	mul r13,r30
-	add r23,r0
+	adiw r28,71-62
+	std Y+63,r17
+	std Y+62,r16
+	sbiw r28,71-62
+	ldd r16,Y+7
+	ldd r17,Y+8
+	mul r16,r18
+	movw r14,r0
+	mul r16,r19
+	add r15,r0
+	mul r17,r18
+	add r15,r0
 	clr r1
-	std Y+52,r23
-	std Y+51,r22
-	ldd r20,Y+3
-	ldd r21,Y+4
-	mul r20,r4
+	adiw r28,67-62
+	std Y+63,r15
+	std Y+62,r14
+	sbiw r28,67-62
+	ldd r14,Y+9
+	ldd r15,Y+10
+	ldd r16,Y+51
+	ldd r17,Y+52
+	mul r14,r16
 	movw r26,r0
-	mul r20,r5
+	mul r14,r17
 	add r27,r0
-	mul r21,r4
+	mul r15,r16
 	add r27,r0
 	clr r1
-	ldd r30,Y+5
-	ldd r31,Y+6
-	ldd r20,Y+31
-	ldd r21,Y+32
-	mul r30,r20
-	movw r22,r0
-	mul r30,r21
-	add r23,r0
-	mul r31,r20
-	add r23,r0
-	clr r1
-	adiw r28,71-62
-	std Y+63,r23
-	std Y+62,r22
-	sbiw r28,71-62
-	ldd r30,Y+7
-	ldd r31,Y+8
-	ldd r20,Y+33
-	ldd r21,Y+34
-	mul r30,r20
-	movw r22,r0
-	mul r30,r21
-	add r23,r0
-	mul r31,r20
-	add r23,r0
-	clr r1
 	adiw r28,69-62
-	std Y+63,r23
-	std Y+62,r22
+	std Y+63,r27
+	std Y+62,r26
 	sbiw r28,69-62
-	ldd r30,Y+9
-	ldd r31,Y+10
-	mul r30,r18
-	movw r20,r0
-	mul r30,r19
-	add r21,r0
-	mul r31,r18
-	add r21,r0
-	clr r1
-	sts k,r24
-	sts k+1,r25
-	sts k+2,r14
-	sts k+3,r15
-	sts k+4,r16
-	sts k+5,r17
-	ldd r31,Y+51
-	sts k+6,r31
-	ldd r22,Y+52
-	sts k+7,r22
-	std Y+54,r27
-	std Y+53,r26
-	sts k+8,r26
-	sts k+9,r27
-	adiw r28,71-62
-	ldd r30,Y+62
-	ldd r31,Y+63
-	sbiw r28,71-62
-	sts k+10,r30
-	sts k+11,r31
-	adiw r28,69-62
-	ldd r22,Y+62
-	ldd r23,Y+63
-	sbiw r28,69-62
-	sts k+12,r22
-	sts k+13,r23
-	sts k+14,r20
-	sts k+15,r21
-	sts res,r24
-	sts res+1,r25
-	sts res+2,r14
-	sts res+3,r15
-	sts res+4,r16
-	sts res+5,r17
-	ldd r26,Y+51
-	sts res+6,r26
-	ldd r27,Y+52
-	sts res+7,r27
-	ldd r26,Y+53
-	sts res+8,r26
-	ldd r27,Y+54
-	sts res+9,r27
-	sts res+10,r30
-	sts res+11,r31
-	sts res+12,r22
-	sts res+13,r23
-	sts res+14,r20
-	sts res+15,r21
-	cpi r24,-36
-	sbci r25,5
-	breq .+2
-	rjmp .L7
-	ldi r31,20
-	cp r14,r31
-	ldi r31,5
-	cpc r15,r31
+	sts k,r30
+	sts k+1,r31
+	sts k+2,r22
+	sts k+3,r23
+	sts k+4,r20
+	sts k+5,r21
+	sts k+6,r24
+	sts k+7,r25
+	sts k+8,r12
+	sts k+9,r13
+	adiw r28,71-63
+	ldd r17,Y+63
+	sbiw r28,71-63
+	sts k+10,r17
+	adiw r28,72-63
+	ldd r26,Y+63
+	sbiw r28,72-63
+	sts k+11,r26
+	adiw r28,67-63
+	ldd r27,Y+63
+	sbiw r28,67-63
+	sts k+12,r27
+	adiw r28,68-63
+	ldd r14,Y+63
+	sbiw r28,68-63
+	sts k+13,r14
+	adiw r28,69-63
+	ldd r15,Y+63
+	sbiw r28,69-63
+	sts k+14,r15
+	adiw r28,70-63
+	ldd r16,Y+63
+	sbiw r28,70-63
+	sts k+15,r16
+	sts res,r30
+	sts res+1,r31
+	sts res+2,r22
+	sts res+3,r23
+	sts res+4,r20
+	sts res+5,r21
+	sts res+6,r24
+	sts res+7,r25
+	sts res+8,r12
+	sts res+9,r13
+	sts res+10,r17
+	sts res+11,r26
+	sts res+12,r27
+	sts res+13,r14
+	sts res+14,r15
+	sts res+15,r16
+	cpi r30,-36
+	sbci r31,5
 	breq .+2
 	rjmp .L7
-	cpi r16,-72
-	sbci r17,11
+	cpi r22,20
+	sbci r23,5
 	breq .+2
 	rjmp .L7
-	ldd r22,Y+51
-	ldd r23,Y+52
-	cpi r22,112
-	sbci r23,23
+	cpi r20,-72
+	sbci r21,11
 	breq .+2
 	rjmp .L7
+	cpi r24,112
+	sbci r25,23
+	breq .+2
+	rjmp .L7
+	ldd r24,Y+1
+	ldd r25,Y+2
+	ldd r22,Y+27
+	ldd r23,Y+28
+	call __divmodhi4
+	movw r12,r22
 	movw r24,r6
-	ldd r22,Y+1
-	ldd r23,Y+2
+	ldd r22,Y+29
+	ldd r23,Y+30
 	call __divmodhi4
 	movw r14,r22
 	movw r24,r8
-	ldd r22,Y+27
-	ldd r23,Y+28
+	movw r22,r2
 	call __divmodhi4
 	movw r16,r22
 	movw r24,r10
-	movw r22,r2
-	call __divmodhi4
-	std Y+2,r23
-	std Y+1,r22
-	movw r24,r12
-	ldd r22,Y+29
-	ldd r23,Y+30
+	ldd r22,Y+31
+	ldd r23,Y+32
 	call __divmodhi4
 	movw r30,r22
 	ldd r24,Y+3
@@ -458,66 +462,63 @@
 	movw r2,r22
 	ldd r24,Y+5
 	ldd r25,Y+6
-	ldd r22,Y+31
-	ldd r23,Y+32
+	ldd r22,Y+33
+	ldd r23,Y+34
 	call __divmodhi4
 	movw r4,r22
 	ldd r24,Y+7
 	ldd r25,Y+8
-	ldd r22,Y+33
-	ldd r23,Y+34
+	movw r22,r18
 	call __divmodhi4
-	sts res+12,r22
+	mov r18,r22
 	mov r20,r23
 	ldd r24,Y+9
 	ldd r25,Y+10
-	movw r22,r18
+	ldd r22,Y+51
+	ldd r23,Y+52
 	call __divmodhi4
-	sts k,r14
-	sts k+1,r15
-	sts k+2,r16
-	sts k+3,r17
-	ldd r25,Y+1
-	sts k+4,r25
-	ldd r26,Y+2
-	sts k+5,r26
+	sts k,r12
+	sts k+1,r13
+	sts k+2,r14
+	sts k+3,r15
+	sts k+4,r16
+	sts k+5,r17
 	sts k+6,r30
 	sts k+7,r31
 	sts k+8,r2
 	sts k+9,r3
 	sts k+10,r4
 	sts k+11,r5
-	lds r18,res+12
 	sts k+12,r18
 	sts k+13,r20
 	sts k+14,r22
 	sts k+15,r23
-	sts res,r14
-	sts res+1,r15
-	sts res+2,r16
-	sts res+3,r17
-	sts res+4,r25
-	sts res+5,r26
+	sts res,r12
+	sts res+1,r13
+	sts res+2,r14
+	sts res+3,r15
+	sts res+4,r16
+	sts res+5,r17
 	sts res+6,r30
 	sts res+7,r31
 	sts res+8,r2
 	sts res+9,r3
 	sts res+10,r4
 	sts res+11,r5
+	sts res+12,r18
 	sts res+13,r20
 	sts res+14,r22
 	sts res+15,r23
-	ldi r25,15
-	cp r14,r25
-	cpc r15,__zero_reg__
+	ldi r18,15
+	cp r12,r18
+	cpc r13,__zero_reg__
 	breq .+2
 	rjmp .L7
-	cpi r16,7
-	cpc r17,__zero_reg__
+	ldi r20,7
+	cp r14,r20
+	cpc r15,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldd r16,Y+1
-	ldd r17,Y+2
 	cpi r16,7
 	cpc r17,__zero_reg__
 	breq .+2
@@ -525,173 +526,162 @@
 	sbiw r30,6
 	breq .+2
 	rjmp .L7
-	mov r18,r6
-	mov r30,r7
-	mov r16,r8
+	mov r20,r6
 	ldd r15,Y+26
-	and r15,r19
-	ldd r20,Y+35
-	and r18,r20
-	std Y+53,r18
-	ldd r31,Y+36
-	and r30,r31
-	std Y+54,r30
-	ldd r17,Y+37
-	and r16,r17
-	std Y+55,r16
-	ldd r18,Y+14
-	ldd r20,Y+38
-	and r18,r20
-	std Y+56,r18
-	ldd r30,Y+15
-	ldd r31,Y+39
-	and r30,r31
-	std Y+57,r30
-	ldd r16,Y+16
-	ldd r17,Y+40
-	and r16,r17
-	std Y+58,r16
-	ldd r18,Y+17
-	ldd r20,Y+41
-	and r18,r20
-	std Y+59,r18
-	ldd r30,Y+18
-	ldd r31,Y+42
-	and r30,r31
-	std Y+60,r30
-	ldd r16,Y+19
-	ldd r17,Y+43
-	and r16,r17
-	std Y+61,r16
-	ldd r18,Y+20
-	ldd r20,Y+44
-	and r18,r20
-	std Y+62,r18
-	ldd r30,Y+21
-	ldd r31,Y+45
-	and r30,r31
-	std Y+63,r30
-	ldd r16,Y+22
-	ldd r17,Y+46
-	and r16,r17
+	ldd r30,Y+50
+	and r15,r30
+	ldd r14,Y+11
+	ldd r16,Y+35
+	and r14,r16
+	std Y+51,r14
+	ldd r17,Y+12
+	ldd r18,Y+36
+	and r17,r18
+	std Y+52,r17
+	ldd r30,Y+37
+	and r20,r30
+	std Y+53,r20
+	ldd r14,Y+14
+	ldd r16,Y+38
+	and r14,r16
+	std Y+54,r14
+	ldd r17,Y+15
+	ldd r18,Y+39
+	and r17,r18
+	std Y+55,r17
+	ldd r20,Y+16
+	ldd r30,Y+40
+	and r20,r30
+	std Y+56,r20
+	ldd r14,Y+17
+	ldd r16,Y+41
+	and r14,r16
+	std Y+57,r14
+	ldd r17,Y+18
+	ldd r18,Y+42
+	and r17,r18
+	std Y+58,r17
+	ldd r20,Y+19
+	ldd r30,Y+43
+	and r20,r30
+	std Y+59,r20
+	ldd r14,Y+20
+	ldd r16,Y+44
+	and r14,r16
+	std Y+60,r14
+	ldd r17,Y+21
+	ldd r18,Y+45
+	and r17,r18
+	std Y+61,r17
+	ldd r20,Y+22
+	ldd r30,Y+46
+	and r20,r30
+	std Y+62,r20
+	ldd r14,Y+23
+	ldd r16,Y+47
+	and r14,r16
+	std Y+63,r14
+	ldd r17,Y+24
+	ldd r18,Y+48
+	and r17,r18
 	adiw r28,64-63
-	std Y+63,r16
+	std Y+63,r17
 	sbiw r28,64-63
-	ldd r18,Y+23
-	ldd r20,Y+47
-	and r18,r20
+	ldd r20,Y+25
+	ldd r30,Y+49
+	and r20,r30
 	adiw r28,65-63
-	std Y+63,r18
+	std Y+63,r20
 	sbiw r28,65-63
-	ldd r30,Y+24
-	ldd r31,Y+48
-	and r30,r31
-	adiw r28,66-63
-	std Y+63,r30
-	sbiw r28,66-63
-	ldd r16,Y+25
-	ldd r17,Y+49
-	and r16,r17
-	adiw r28,67-63
-	std Y+63,r16
-	sbiw r28,67-63
-	ldd r18,Y+53
-	sts k,r18
-	ldd r20,Y+54
-	sts k+1,r20
-	ldd r21,Y+55
-	sts k+2,r21
-	ldd r22,Y+56
-	sts k+3,r22
-	ldd r23,Y+57
-	sts k+4,r23
+	ldd r14,Y+51
+	sts k,r14
+	ldd r16,Y+52
+	sts k+1,r16
+	ldd r17,Y+53
+	sts k+2,r17
+	ldd r18,Y+54
+	sts k+3,r18
+	ldd r20,Y+55
+	sts k+4,r20
+	ldd r21,Y+56
+	sts k+5,r21
+	ldd r22,Y+57
+	sts k+6,r22
 	ldd r24,Y+58
-	sts k+5,r24
+	sts k+7,r24
 	ldd r25,Y+59
-	sts k+6,r25
+	sts k+8,r25
 	ldd r26,Y+60
-	sts k+7,r26
+	sts k+9,r26
 	ldd r27,Y+61
-	sts k+8,r27
+	sts k+10,r27
 	ldd r30,Y+62
-	sts k+9,r30
-	ldd r31,Y+63
-	sts k+10,r31
+	sts k+11,r30
+	ldd r14,Y+63
+	sts k+12,r14
 	adiw r28,64-63
 	ldd r16,Y+63
 	sbiw r28,64-63
-	sts k+11,r16
+	sts k+13,r16
 	adiw r28,65-63
 	ldd r17,Y+63
 	sbiw r28,65-63
-	sts k+12,r17
-	adiw r28,66-63
-	ldd r18,Y+63
-	sbiw r28,66-63
-	sts k+13,r18
-	adiw r28,67-63
-	ldd r20,Y+63
-	sbiw r28,67-63
-	sts k+14,r20
+	sts k+14,r17
 	sts k+15,r15
+	ldd r18,Y+51
+	sts res,r18
+	ldd r20,Y+52
+	sts res+1,r20
 	ldd r21,Y+53
-	sts res,r21
+	sts res+2,r21
 	ldd r22,Y+54
-	sts res+1,r22
-	ldd r23,Y+55
-	sts res+2,r23
-	ldd r24,Y+56
-	sts res+3,r24
-	ldd r25,Y+57
-	sts res+4,r25
-	ldd r26,Y+58
-	sts res+5,r26
-	ldd r27,Y+59
-	sts res+6,r27
-	ldd r30,Y+60
-	sts res+7,r30
-	ldd r31,Y+61
-	sts res+8,r31
-	ldd r16,Y+62
-	sts res+9,r16
-	ldd r17,Y+63
-	sts res+10,r17
-	adiw r28,64-63
+	sts res+3,r22
+	ldd r24,Y+55
+	sts res+4,r24
+	ldd r25,Y+56
+	sts res+5,r25
+	ldd r26,Y+57
+	sts res+6,r26
+	ldd r27,Y+58
+	sts res+7,r27
+	ldd r30,Y+59
+	sts res+8,r30
+	ldd r14,Y+60
+	sts res+9,r14
+	ldd r16,Y+61
+	sts res+10,r16
+	ldd r17,Y+62
+	sts res+11,r17
 	ldd r18,Y+63
+	sts res+12,r18
+	adiw r28,64-63
+	ldd r20,Y+63
 	sbiw r28,64-63
-	sts res+11,r18
+	sts res+13,r20
 	adiw r28,65-63
-	ldd r20,Y+63
-	sbiw r28,65-63
-	sts res+12,r20
-	adiw r28,66-63
 	ldd r21,Y+63
-	sbiw r28,66-63
-	sts res+13,r21
-	adiw r28,67-63
-	ldd r22,Y+63
-	sbiw r28,67-63
-	sts res+14,r22
+	sbiw r28,65-63
+	sts res+14,r21
 	sts res+15,r15
-	ldd r14,Y+59
-	ldd r15,Y+60
-	ldd r4,Y+57
-	ldd r5,Y+58
-	ldd r2,Y+55
-	ldd r3,Y+56
-	ldd r24,Y+53
-	ldd r25,Y+54
+	ldd r14,Y+57
+	ldd r15,Y+58
+	ldd r12,Y+55
+	ldd r13,Y+56
+	ldd r4,Y+53
+	ldd r5,Y+54
+	ldd r24,Y+51
+	ldd r25,Y+52
 	sbiw r24,2
 	breq .+2
 	rjmp .L7
 	ldi r25,4
-	cp r2,r25
-	cpc r3,__zero_reg__
+	cp r4,r25
+	cpc r5,__zero_reg__
 	breq .+2
 	rjmp .L7
 	ldi r26,20
-	cp r4,r26
-	cpc r5,__zero_reg__
+	cp r12,r26
+	cpc r13,__zero_reg__
 	breq .+2
 	rjmp .L7
 	ldi r27,8
@@ -699,608 +689,574 @@
 	cpc r15,__zero_reg__
 	breq .+2
 	rjmp .L7
-	mov r31,r6
-	mov r17,r7
-	mov r20,r8
+	mov r20,r6
 	ldd r15,Y+26
-	or r15,r19
+	ldd r30,Y+50
+	or r15,r30
+	ldd r14,Y+11
 	ldd r16,Y+35
-	or r31,r16
-	std Y+53,r31
+	or r14,r16
+	std Y+51,r14
+	ldd r17,Y+12
 	ldd r18,Y+36
 	or r17,r18
-	std Y+54,r17
+	std Y+52,r17
 	ldd r30,Y+37
 	or r20,r30
-	std Y+55,r20
-	ldd r31,Y+14
+	std Y+53,r20
+	ldd r14,Y+14
 	ldd r16,Y+38
-	or r31,r16
-	std Y+56,r31
+	or r14,r16
+	std Y+54,r14
 	ldd r17,Y+15
 	ldd r18,Y+39
 	or r17,r18
-	std Y+57,r17
+	std Y+55,r17
 	ldd r20,Y+16
 	ldd r30,Y+40
 	or r20,r30
-	std Y+58,r20
-	ldd r31,Y+17
+	std Y+56,r20
+	ldd r14,Y+17
 	ldd r16,Y+41
-	or r31,r16
-	std Y+59,r31
+	or r14,r16
+	std Y+57,r14
 	ldd r17,Y+18
 	ldd r18,Y+42
 	or r17,r18
-	std Y+60,r17
+	std Y+58,r17
 	ldd r20,Y+19
 	ldd r30,Y+43
 	or r20,r30
-	std Y+61,r20
-	ldd r31,Y+20
+	std Y+59,r20
+	ldd r14,Y+20
 	ldd r16,Y+44
-	or r31,r16
-	std Y+62,r31
+	or r14,r16
+	std Y+60,r14
 	ldd r17,Y+21
 	ldd r18,Y+45
 	or r17,r18
-	std Y+63,r17
+	std Y+61,r17
 	ldd r20,Y+22
 	ldd r30,Y+46
 	or r20,r30
-	adiw r28,64-63
-	std Y+63,r20
-	sbiw r28,64-63
-	ldd r31,Y+23
+	std Y+62,r20
+	ldd r14,Y+23
 	ldd r16,Y+47
-	or r31,r16
-	adiw r28,65-63
-	std Y+63,r31
-	sbiw r28,65-63
+	or r14,r16
+	std Y+63,r14
 	ldd r17,Y+24
 	ldd r18,Y+48
 	or r17,r18
-	adiw r28,66-63
+	adiw r28,64-63
 	std Y+63,r17
-	sbiw r28,66-63
+	sbiw r28,64-63
 	ldd r20,Y+25
 	ldd r30,Y+49
 	or r20,r30
-	adiw r28,67-63
+	adiw r28,65-63
 	std Y+63,r20
-	sbiw r28,67-63
-	ldd r31,Y+53
-	sts k,r31
-	ldd r16,Y+54
+	sbiw r28,65-63
+	ldd r14,Y+51
+	sts k,r14
+	ldd r16,Y+52
 	sts k+1,r16
-	ldd r17,Y+55
+	ldd r17,Y+53
 	sts k+2,r17
-	ldd r18,Y+56
+	ldd r18,Y+54
 	sts k+3,r18
-	ldd r20,Y+57
+	ldd r20,Y+55
 	sts k+4,r20
-	ldd r21,Y+58
+	ldd r21,Y+56
 	sts k+5,r21
-	ldd r22,Y+59
+	ldd r22,Y+57
 	sts k+6,r22
-	ldd r23,Y+60
-	sts k+7,r23
-	ldd r24,Y+61
-	sts k+8,r24
-	ldd r25,Y+62
-	sts k+9,r25
-	ldd r26,Y+63
-	sts k+10,r26
+	ldd r24,Y+58
+	sts k+7,r24
+	ldd r25,Y+59
+	sts k+8,r25
+	ldd r26,Y+60
+	sts k+9,r26
+	ldd r27,Y+61
+	sts k+10,r27
+	ldd r30,Y+62
+	sts k+11,r30
+	ldd r14,Y+63
+	sts k+12,r14
 	adiw r28,64-63
-	ldd r27,Y+63
+	ldd r16,Y+63
 	sbiw r28,64-63
-	sts k+11,r27
+	sts k+13,r16
 	adiw r28,65-63
-	ldd r30,Y+63
+	ldd r17,Y+63
 	sbiw r28,65-63
-	sts k+12,r30
-	adiw r28,66-63
-	ldd r31,Y+63
-	sbiw r28,66-63
-	sts k+13,r31
-	adiw r28,67-63
-	ldd r16,Y+63
-	sbiw r28,67-63
-	sts k+14,r16
+	sts k+14,r17
 	sts k+15,r15
-	ldd r17,Y+53
-	sts res,r17
-	ldd r18,Y+54
-	sts res+1,r18
-	ldd r20,Y+55
-	sts res+2,r20
-	ldd r21,Y+56
-	sts res+3,r21
-	ldd r22,Y+57
-	sts res+4,r22
-	ldd r23,Y+58
-	sts res+5,r23
-	ldd r24,Y+59
-	sts res+6,r24
-	ldd r25,Y+60
-	sts res+7,r25
-	ldd r26,Y+61
-	sts res+8,r26
-	ldd r27,Y+62
-	sts res+9,r27
-	ldd r30,Y+63
-	sts res+10,r30
+	ldd r18,Y+51
+	sts res,r18
+	ldd r20,Y+52
+	sts res+1,r20
+	ldd r21,Y+53
+	sts res+2,r21
+	ldd r22,Y+54
+	sts res+3,r22
+	ldd r24,Y+55
+	sts res+4,r24
+	ldd r25,Y+56
+	sts res+5,r25
+	ldd r26,Y+57
+	sts res+6,r26
+	ldd r27,Y+58
+	sts res+7,r27
+	ldd r30,Y+59
+	sts res+8,r30
+	ldd r14,Y+60
+	sts res+9,r14
+	ldd r16,Y+61
+	sts res+10,r16
+	ldd r17,Y+62
+	sts res+11,r17
+	ldd r18,Y+63
+	sts res+12,r18
 	adiw r28,64-63
-	ldd r31,Y+63
+	ldd r20,Y+63
 	sbiw r28,64-63
-	sts res+11,r31
+	sts res+13,r20
 	adiw r28,65-63
-	ldd r16,Y+63
+	ldd r21,Y+63
 	sbiw r28,65-63
-	sts res+12,r16
-	adiw r28,66-63
-	ldd r17,Y+63
-	sbiw r28,66-63
-	sts res+13,r17
-	adiw r28,67-63
-	ldd r18,Y+63
-	sbiw r28,67-63
-	sts res+14,r18
+	sts res+14,r21
 	sts res+15,r15
-	ldd r14,Y+59
-	ldd r15,Y+60
-	ldd r4,Y+57
-	ldd r5,Y+58
-	ldd r2,Y+55
-	ldd r3,Y+56
-	ldd r20,Y+53
-	ldd r21,Y+54
-	cpi r20,-98
-	cpc r21,__zero_reg__
+	ldd r14,Y+57
+	ldd r15,Y+58
+	ldd r12,Y+55
+	ldd r13,Y+56
+	ldd r4,Y+53
+	ldd r5,Y+54
+	ldd r24,Y+51
+	ldd r25,Y+52
+	cpi r24,-98
+	cpc r25,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldi r21,109
-	cp r2,r21
-	cpc r3,__zero_reg__
+	ldi r25,109
+	cp r4,r25
+	cpc r5,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldi r22,-106
-	cp r4,r22
-	cpc r5,__zero_reg__
+	ldi r26,-106
+	cp r12,r26
+	cpc r13,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldi r23,-34
-	cp r14,r23
+	ldi r27,-34
+	cp r14,r27
 	cpc r15,__zero_reg__
 	breq .+2
 	rjmp .L7
 	ldd r15,Y+26
 	ldd r30,Y+50
 	eor r15,r30
-	ldd r31,Y+11
+	ldd r14,Y+11
 	ldd r16,Y+35
-	eor r31,r16
-	std Y+53,r31
+	eor r14,r16
+	std Y+51,r14
 	ldd r17,Y+12
 	ldd r18,Y+36
 	eor r17,r18
-	std Y+54,r17
+	std Y+52,r17
 	ldd r20,Y+13
 	ldd r30,Y+37
 	eor r20,r30
-	std Y+55,r20
-	ldd r31,Y+14
+	std Y+53,r20
+	ldd r14,Y+14
 	ldd r16,Y+38
-	eor r31,r16
-	std Y+56,r31
+	eor r14,r16
+	std Y+54,r14
 	ldd r17,Y+15
 	ldd r18,Y+39
 	eor r17,r18
-	std Y+57,r17
+	std Y+55,r17
 	ldd r20,Y+16
 	ldd r30,Y+40
 	eor r20,r30
-	std Y+58,r20
-	ldd r31,Y+17
+	std Y+56,r20
+	ldd r14,Y+17
 	ldd r16,Y+41
-	eor r31,r16
-	std Y+59,r31
+	eor r14,r16
+	std Y+57,r14
 	ldd r17,Y+18
 	ldd r18,Y+42
 	eor r17,r18
-	std Y+60,r17
+	std Y+58,r17
 	ldd r20,Y+19
 	ldd r30,Y+43
 	eor r20,r30
-	std Y+61,r20
-	ldd r31,Y+20
+	std Y+59,r20
+	ldd r14,Y+20
 	ldd r16,Y+44
-	eor r31,r16
-	std Y+62,r31
+	eor r14,r16
+	std Y+60,r14
 	ldd r17,Y+21
 	ldd r18,Y+45
 	eor r17,r18
-	std Y+63,r17
+	std Y+61,r17
 	ldd r20,Y+22
 	ldd r30,Y+46
 	eor r20,r30
-	adiw r28,64-63
-	std Y+63,r20
-	sbiw r28,64-63
-	ldd r31,Y+23
+	std Y+62,r20
+	ldd r14,Y+23
 	ldd r16,Y+47
-	eor r31,r16
-	adiw r28,65-63
-	std Y+63,r31
-	sbiw r28,65-63
+	eor r14,r16
+	std Y+63,r14
 	ldd r17,Y+24
 	ldd r18,Y+48
 	eor r17,r18
-	adiw r28,66-63
+	adiw r28,64-63
 	std Y+63,r17
-	sbiw r28,66-63
+	sbiw r28,64-63
 	ldd r20,Y+25
 	ldd r30,Y+49
 	eor r20,r30
-	adiw r28,67-63
+	adiw r28,65-63
 	std Y+63,r20
-	sbiw r28,67-63
-	ldd r31,Y+53
-	sts k,r31
-	ldd r16,Y+54
+	sbiw r28,65-63
+	ldd r14,Y+51
+	sts k,r14
+	ldd r16,Y+52
 	sts k+1,r16
-	ldd r17,Y+55
+	ldd r17,Y+53
 	sts k+2,r17
-	ldd r18,Y+56
+	ldd r18,Y+54
 	sts k+3,r18
-	ldd r20,Y+57
+	ldd r20,Y+55
 	sts k+4,r20
-	ldd r21,Y+58
+	ldd r21,Y+56
 	sts k+5,r21
-	ldd r22,Y+59
+	ldd r22,Y+57
 	sts k+6,r22
-	ldd r23,Y+60
-	sts k+7,r23
-	ldd r24,Y+61
-	sts k+8,r24
-	ldd r25,Y+62
-	sts k+9,r25
-	ldd r26,Y+63
-	sts k+10,r26
+	ldd r24,Y+58
+	sts k+7,r24
+	ldd r25,Y+59
+	sts k+8,r25
+	ldd r26,Y+60
+	sts k+9,r26
+	ldd r27,Y+61
+	sts k+10,r27
+	ldd r30,Y+62
+	sts k+11,r30
+	ldd r14,Y+63
+	sts k+12,r14
 	adiw r28,64-63
-	ldd r27,Y+63
+	ldd r16,Y+63
 	sbiw r28,64-63
-	sts k+11,r27
+	sts k+13,r16
 	adiw r28,65-63
-	ldd r30,Y+63
+	ldd r17,Y+63
 	sbiw r28,65-63
-	sts k+12,r30
-	adiw r28,66-63
-	ldd r31,Y+63
-	sbiw r28,66-63
-	sts k+13,r31
-	adiw r28,67-63
-	ldd r16,Y+63
-	sbiw r28,67-63
-	sts k+14,r16
+	sts k+14,r17
 	sts k+15,r15
-	ldd r17,Y+53
-	sts res,r17
-	ldd r18,Y+54
-	sts res+1,r18
-	ldd r20,Y+55
-	sts res+2,r20
-	ldd r21,Y+56
-	sts res+3,r21
-	ldd r22,Y+57
-	sts res+4,r22
-	ldd r23,Y+58
-	sts res+5,r23
-	ldd r24,Y+59
-	sts res+6,r24
-	ldd r25,Y+60
-	sts res+7,r25
-	ldd r26,Y+61
-	sts res+8,r26
-	ldd r27,Y+62
-	sts res+9,r27
-	ldd r30,Y+63
-	sts res+10,r30
+	ldd r18,Y+51
+	sts res,r18
+	ldd r20,Y+52
+	sts res+1,r20
+	ldd r21,Y+53
+	sts res+2,r21
+	ldd r22,Y+54
+	sts res+3,r22
+	ldd r24,Y+55
+	sts res+4,r24
+	ldd r25,Y+56
+	sts res+5,r25
+	ldd r26,Y+57
+	sts res+6,r26
+	ldd r27,Y+58
+	sts res+7,r27
+	ldd r30,Y+59
+	sts res+8,r30
+	ldd r14,Y+60
+	sts res+9,r14
+	ldd r16,Y+61
+	sts res+10,r16
+	ldd r17,Y+62
+	sts res+11,r17
+	ldd r18,Y+63
+	sts res+12,r18
 	adiw r28,64-63
-	ldd r31,Y+63
+	ldd r20,Y+63
 	sbiw r28,64-63
-	sts res+11,r31
+	sts res+13,r20
 	adiw r28,65-63
-	ldd r16,Y+63
+	ldd r21,Y+63
 	sbiw r28,65-63
-	sts res+12,r16
-	adiw r28,66-63
-	ldd r17,Y+63
-	sbiw r28,66-63
-	sts res+13,r17
-	adiw r28,67-63
-	ldd r18,Y+63
-	sbiw r28,67-63
-	sts res+14,r18
+	sts res+14,r21
 	sts res+15,r15
-	ldd r14,Y+59
-	ldd r15,Y+60
-	ldd r4,Y+57
-	ldd r5,Y+58
-	ldd r2,Y+55
-	ldd r3,Y+56
-	ldd r20,Y+53
-	ldd r21,Y+54
-	cpi r20,-100
-	cpc r21,__zero_reg__
+	ldd r14,Y+57
+	ldd r15,Y+58
+	ldd r12,Y+55
+	ldd r13,Y+56
+	ldd r4,Y+53
+	ldd r5,Y+54
+	ldd r24,Y+51
+	ldd r25,Y+52
+	cpi r24,-100
+	cpc r25,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldi r21,105
-	cp r2,r21
-	cpc r3,__zero_reg__
+	ldi r25,105
+	cp r4,r25
+	cpc r5,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldi r22,-126
-	cp r4,r22
-	cpc r5,__zero_reg__
+	ldi r26,-126
+	cp r12,r26
+	cpc r13,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldi r23,-42
-	cp r14,r23
+	ldi r27,-42
+	cp r14,r27
 	cpc r15,__zero_reg__
 	breq .+2
 	rjmp .L7
-	clr r16
-	clr r17
-	sub r16,r6
-	sbc r17,r7
-	clr r20
-	clr r21
-	sub r20,r8
-	sbc r21,r9
-	clr r18
-	clr r19
-	sub r18,r10
-	sbc r19,r11
-	clr r24
-	clr r25
-	sub r24,r12
-	sbc r25,r13
-	ldd r30,Y+3
-	ldd r31,Y+4
-	neg r31
-	neg r30
-	sbc r31,__zero_reg__
-	std Y+2,r31
-	std Y+1,r30
-	ldd r26,Y+5
-	ldd r27,Y+6
-	neg r27
-	neg r26
-	sbc r27,__zero_reg__
-	ldd r30,Y+7
-	ldd r31,Y+8
-	neg r31
-	neg r30
-	sbc r31,__zero_reg__
-	ldd r22,Y+9
-	ldd r23,Y+10
+	ldd r16,Y+1
+	ldd r17,Y+2
+	neg r17
+	neg r16
+	sbc r17,__zero_reg__
+	clr r14
+	clr r15
+	sub r14,r6
+	sbc r15,r7
+	std Y+28,r15
+	std Y+27,r14
+	clr r26
+	clr r27
+	sub r26,r8
+	sbc r27,r9
+	clr r30
+	clr r31
+	sub r30,r10
+	sbc r31,r11
+	ldd r22,Y+3
+	ldd r23,Y+4
 	neg r23
 	neg r22
 	sbc r23,__zero_reg__
+	ldd r20,Y+5
+	ldd r21,Y+6
+	neg r21
+	neg r20
+	sbc r21,__zero_reg__
+	ldd r18,Y+7
+	ldd r19,Y+8
+	neg r19
+	neg r18
+	sbc r19,__zero_reg__
+	ldd r24,Y+9
+	ldd r25,Y+10
+	neg r25
+	neg r24
+	sbc r25,__zero_reg__
 	sts k,r16
 	sts k+1,r17
-	sts k+2,r20
-	sts k+3,r21
-	sts k+4,r18
-	sts k+5,r19
-	sts k+6,r24
-	sts k+7,r25
-	ldd r15,Y+1
-	sts k+8,r15
-	ldd r15,Y+2
-	sts k+9,r15
-	sts k+10,r26
-	sts k+11,r27
-	sts k+12,r30
-	sts k+13,r31
-	sts k+14,r22
-	sts k+15,r23
+	sts k+2,r14
+	sts k+3,r15
+	sts k+4,r26
+	sts k+5,r27
+	sts k+6,r30
+	sts k+7,r31
+	sts k+8,r22
+	sts k+9,r23
+	sts k+10,r20
+	sts k+11,r21
+	sts k+12,r18
+	sts k+13,r19
+	sts k+14,r24
+	sts k+15,r25
 	sts res,r16
 	sts res+1,r17
-	sts res+2,r20
-	sts res+3,r21
-	sts res+4,r18
-	sts res+5,r19
-	sts res+6,r24
-	sts res+7,r25
-	ldd r16,Y+1
-	sts res+8,r16
-	sts res+9,r15
-	sts res+10,r26
-	sts res+11,r27
-	sts res+12,r30
-	sts res+13,r31
-	sts res+14,r22
-	sts res+15,r23
-	ldi r18,-106
-	cp r6,r18
+	ldd r15,Y+27
+	sts res+2,r15
+	ldd r16,Y+28
+	sts res+3,r16
+	sts res+4,r26
+	sts res+5,r27
+	sts res+6,r30
+	sts res+7,r31
+	sts res+8,r22
+	sts res+9,r23
+	sts res+10,r20
+	sts res+11,r21
+	sts res+12,r18
+	sts res+13,r19
+	sts res+14,r24
+	sts res+15,r25
+	ldd r20,Y+1
+	ldd r21,Y+2
+	cpi r20,-106
+	cpc r21,__zero_reg__
+	breq .+2
+	rjmp .L7
+	ldi r21,100
+	cp r6,r21
 	cpc r7,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldi r20,100
-	cp r8,r20
+	ldi r22,-106
+	cp r8,r22
 	cpc r9,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldi r21,-106
-	cp r10,r21
+	ldi r24,-56
+	cp r10,r24
 	cpc r11,__zero_reg__
 	breq .+2
 	rjmp .L7
-	ldi r22,-56
-	cp r12,r22
-	cpc r13,__zero_reg__
-	breq .+2
-	rjmp .L7
 	ldd r15,Y+26
 	com r15
 	ldd r30,Y+11
 	com r30
-	std Y+53,r30
-	ldd r31,Y+12
-	com r31
-	std Y+54,r31
+	std Y+51,r30
+	ldd r14,Y+12
+	com r14
+	std Y+52,r14
 	ldd r16,Y+13
 	com r16
-	std Y+55,r16
+	std Y+53,r16
 	ldd r17,Y+14
 	com r17
-	std Y+56,r17
+	std Y+54,r17
 	ldd r18,Y+15
 	com r18
-	std Y+57,r18
+	std Y+55,r18
 	ldd r20,Y+16
 	com r20
-	std Y+58,r20
+	std Y+56,r20
 	ldd r30,Y+17
 	com r30
-	std Y+59,r30
-	ldd r31,Y+18
-	com r31
-	std Y+60,r31
+	std Y+57,r30
+	ldd r14,Y+18
+	com r14
+	std Y+58,r14
 	ldd r16,Y+19
 	com r16
-	std Y+61,r16
+	std Y+59,r16
 	ldd r17,Y+20
 	com r17
-	std Y+62,r17
+	std Y+60,r17
 	ldd r18,Y+21
 	com r18
-	std Y+63,r18
+	std Y+61,r18
 	ldd r20,Y+22
 	com r20
-	adiw r28,64-63
-	std Y+63,r20
-	sbiw r28,64-63
+	std Y+62,r20
 	ldd r30,Y+23
 	com r30
-	adiw r28,65-63
 	std Y+63,r30
-	sbiw r28,65-63
-	ldd r31,Y+24
-	com r31
-	adiw r28,66-63
-	std Y+63,r31
-	sbiw r28,66-63
+	ldd r14,Y+24
+	com r14
+	adiw r28,64-63
+	std Y+63,r14
+	sbiw r28,64-63
 	ldd r16,Y+25
 	com r16
-	adiw r28,67-63
+	adiw r28,65-63
 	std Y+63,r16
-	sbiw r28,67-63
-	ldd r17,Y+53
+	sbiw r28,65-63
+	ldd r17,Y+51
 	sts k,r17
-	ldd r18,Y+54
+	ldd r18,Y+52
 	sts k+1,r18
-	ldd r20,Y+55
+	ldd r20,Y+53
 	sts k+2,r20
-	ldd r21,Y+56
+	ldd r21,Y+54
 	sts k+3,r21
-	ldd r22,Y+57
+	ldd r22,Y+55
 	sts k+4,r22
-	ldd r23,Y+58
-	sts k+5,r23
-	ldd r24,Y+59
-	sts k+6,r24
-	ldd r25,Y+60
-	sts k+7,r25
-	ldd r26,Y+61
-	sts k+8,r26
-	ldd r27,Y+62
-	sts k+9,r27
-	ldd r30,Y+63
-	sts k+10,r30
+	ldd r24,Y+56
+	sts k+5,r24
+	ldd r25,Y+57
+	sts k+6,r25
+	ldd r26,Y+58
+	sts k+7,r26
+	ldd r27,Y+59
+	sts k+8,r27
+	ldd r30,Y+60
+	sts k+9,r30
+	ldd r14,Y+61
+	sts k+10,r14
+	ldd r16,Y+62
+	sts k+11,r16
+	ldd r17,Y+63
+	sts k+12,r17
 	adiw r28,64-63
-	ldd r31,Y+63
+	ldd r18,Y+63
 	sbiw r28,64-63
-	sts k+11,r31
+	sts k+13,r18
 	adiw r28,65-63
-	ldd r16,Y+63
+	ldd r20,Y+63
 	sbiw r28,65-63
-	sts k+12,r16
-	adiw r28,66-63
-	ldd r17,Y+63
-	sbiw r28,66-63
-	sts k+13,r17
-	adiw r28,67-63
-	ldd r18,Y+63
-	sbiw r28,67-63
-	sts k+14,r18
+	sts k+14,r20
 	sts k+15,r15
-	ldd r20,Y+53
-	sts res,r20
-	ldd r21,Y+54
-	sts res+1,r21
-	ldd r22,Y+55
-	sts res+2,r22
-	ldd r23,Y+56
-	sts res+3,r23
-	ldd r24,Y+57
-	sts res+4,r24
-	ldd r25,Y+58
-	sts res+5,r25
-	ldd r26,Y+59
-	sts res+6,r26
-	ldd r27,Y+60
-	sts res+7,r27
-	ldd r30,Y+61
-	sts res+8,r30
-	ldd r31,Y+62
-	sts res+9,r31
-	ldd r16,Y+63
-	sts res+10,r16
+	ldd r21,Y+51
+	sts res,r21
+	ldd r22,Y+52
+	sts res+1,r22
+	ldd r24,Y+53
+	sts res+2,r24
+	ldd r25,Y+54
+	sts res+3,r25
+	ldd r26,Y+55
+	sts res+4,r26
+	ldd r27,Y+56
+	sts res+5,r27
+	ldd r30,Y+57
+	sts res+6,r30
+	ldd r14,Y+58
+	sts res+7,r14
+	ldd r16,Y+59
+	sts res+8,r16
+	ldd r17,Y+60
+	sts res+9,r17
+	ldd r18,Y+61
+	sts res+10,r18
+	ldd r20,Y+62
+	sts res+11,r20
+	ldd r21,Y+63
+	sts res+12,r21
 	adiw r28,64-63
-	ldd r17,Y+63
+	ldd r22,Y+63
 	sbiw r28,64-63
-	sts res+11,r17
+	sts res+13,r22
 	adiw r28,65-63
-	ldd r18,Y+63
+	ldd r24,Y+63
 	sbiw r28,65-63
-	sts res+12,r18
-	adiw r28,66-63
-	ldd r20,Y+63
-	sbiw r28,66-63
-	sts res+13,r20
-	adiw r28,67-63
-	ldd r21,Y+63
-	sbiw r28,67-63
-	sts res+14,r21
+	sts res+14,r24
 	sts res+15,r15
-	ldd r14,Y+59
-	ldd r15,Y+60
-	ldd r12,Y+57
-	ldd r13,Y+58
-	ldd r10,Y+55
-	ldd r11,Y+56
-	ldd r22,Y+53
-	ldd r23,Y+54
-	cpi r22,105
-	sbci r23,-1
+	ldd r14,Y+57
+	ldd r15,Y+58
+	ldd r12,Y+55
+	ldd r13,Y+56
+	ldd r10,Y+53
+	ldd r11,Y+54
+	ldd r26,Y+51
+	ldd r27,Y+52
+	cpi r26,105
+	sbci r27,-1
 	brne .L7
-	ldi r25,-101
-	cp r10,r25
-	ldi r25,-1
-	cpc r11,r25
+	ldi r16,-101
+	cp r10,r16
+	ldi r16,-1
+	cpc r11,r16
 	brne .L7
-	ldi r26,105
-	cp r12,r26
-	ldi r26,-1
-	cpc r13,r26
+	ldi r17,105
+	cp r12,r17
+	ldi r17,-1
+	cpc r13,r17
 	brne .L7
-	ldi r27,55
-	cp r14,r27
-	ldi r27,-1
-	cpc r15,r27
+	ldi r18,55
+	cp r14,r18
+	ldi r18,-1
+	cpc r15,r18
 	brne .L7
 	ldi r25,0
 	ldi r24,0
