DIR ddd1a2b8d6d172ac7620c871697c1c7b
--- 991216-1.c.{atmega128}.{s}.{vanilla}.s	2020-08-05 18:55:36.724769125 +0000
+++ 991216-1.c.{atmega128}.{s}.{ccmode}.s	2020-08-05 18:55:36.520766772 +0000
@@ -288,28 +288,29 @@
 /* frame size = 0 */
 /* stack size = 18 */
 .L__stack_usage = 18
-	movw r26,r24
-	movw r6,r22
-	movw r30,r20
-	movw r4,r18
+	movw r6,r24
+	movw r26,r22
+	movw r4,r20
+	movw r2,r18
+	movw r30,r16
 	movw r18,r8
 	movw r20,r10
 	movw r22,r12
 	movw r24,r14
-	sbiw r26,1
+	dec r6
+	or r6,r7
 	brne .L14
-	ldi r26,2
-	cp r6,r26
-	cpc r7,__zero_reg__
+	sbiw r26,2
 	brne .L14
-	sbiw r30,3
-	brne .L14
-	ldi r30,4
-	cp r4,r30
+	ldi r26,3
+	cp r4,r26
 	cpc r5,__zero_reg__
 	brne .L14
-	cpi r16,5
-	cpc r17,__zero_reg__
+	ldi r17,4
+	cp r2,r17
+	cpc r3,__zero_reg__
+	brne .L14
+	sbiw r30,5
 	brne .L14
 	cpi r18,-17
 	sbci r19,-51
@@ -389,8 +390,8 @@
 	cp r8,r24
 	cpc r9,__zero_reg__
 	brne .L17
-	ldi r24,3
-	cp r10,r24
+	ldi r25,3
+	cp r10,r25
 	cpc r11,__zero_reg__
 	brne .L17
 	ldi r24,4
@@ -459,10 +460,10 @@
 /* frame size = 0 */
 /* stack size = 16 */
 .L__stack_usage = 16
-	movw r6,r22
-	movw r4,r20
-	movw r8,r18
-	movw r10,r16
+	movw r8,r22
+	movw r6,r20
+	movw r10,r18
+	movw r4,r16
 	movw r26,r14
 	movw r30,r12
 	ldd r18,Y+19
@@ -476,21 +477,21 @@
 	sbiw r24,1
 	brne .L20
 	ldi r24,2
-	cp r6,r24
-	cpc r7,__zero_reg__
-	brne .L20
-	ldi r24,3
-	cp r4,r24
-	cpc r5,__zero_reg__
-	brne .L20
-	ldi r24,4
 	cp r8,r24
 	cpc r9,__zero_reg__
 	brne .L20
-	ldi r24,5
+	ldi r25,3
+	cp r6,r25
+	cpc r7,__zero_reg__
+	brne .L20
+	ldi r24,4
 	cp r10,r24
 	cpc r11,__zero_reg__
 	brne .L20
+	ldi r25,5
+	cp r4,r25
+	cpc r5,__zero_reg__
+	brne .L20
 	sbiw r26,6
 	brne .L20
 	sbiw r30,7
@@ -535,8 +536,6 @@
 .global	test8
 	.type	test8, @function
 test8:
-	push r2
-	push r3
 	push r4
 	push r5
 	push r6
@@ -557,49 +556,49 @@
 	in r29,__SP_H__
 /* prologue: function */
 /* frame size = 0 */
-/* stack size = 18 */
-.L__stack_usage = 18
-	movw r6,r22
-	movw r4,r20
-	movw r2,r18
-	movw r8,r16
+/* stack size = 16 */
+.L__stack_usage = 16
+	movw r8,r22
+	movw r6,r20
+	movw r4,r18
+	movw r26,r16
 	movw r16,r14
-	movw r26,r12
+	movw r14,r12
 	movw r30,r10
-	ldd r18,Y+21
-	ldd r19,Y+22
-	ldd r20,Y+23
-	ldd r21,Y+24
-	ldd r22,Y+25
-	ldd r23,Y+26
-	ldd r14,Y+27
-	ldd r15,Y+28
+	ldd r18,Y+19
+	ldd r19,Y+20
+	ldd r20,Y+21
+	ldd r21,Y+22
+	ldd r22,Y+23
+	ldd r23,Y+24
+	ldd r12,Y+25
+	ldd r13,Y+26
 	sbiw r24,1
 	brne .L23
 	ldi r24,2
-	cp r6,r24
+	cp r8,r24
+	cpc r9,__zero_reg__
+	brne .L23
+	ldi r25,3
+	cp r6,r25
 	cpc r7,__zero_reg__
 	brne .L23
-	ldi r24,3
+	ldi r24,4
 	cp r4,r24
 	cpc r5,__zero_reg__
 	brne .L23
-	ldi r24,4
-	cp r2,r24
-	cpc r3,__zero_reg__
-	brne .L23
-	ldi r24,5
-	cp r8,r24
-	cpc r9,__zero_reg__
+	sbiw r26,5
 	brne .L23
 	cpi r16,6
 	cpc r17,__zero_reg__
 	brne .L23
-	sbiw r26,7
+	ldi r25,7
+	cp r14,r25
+	cpc r15,__zero_reg__
 	brne .L23
 	sbiw r30,8
 	brne .L23
-	movw r24,r14
+	movw r24,r12
 	cpi r18,-17
 	sbci r19,-51
 	sbci r20,-85
@@ -609,8 +608,8 @@
 	sbci r24,35
 	sbci r25,1
 	brne .L23
-	ldd r24,Y+29
-	ldd r25,Y+30
+	ldd r24,Y+27
+	ldd r25,Y+28
 	cpi r24,85
 	cpc r25,__zero_reg__
 	breq .L22
@@ -634,8 +633,6 @@
 	pop r6
 	pop r5
 	pop r4
-	pop r3
-	pop r2
 	ret
 	.size	test8, .-test8
 	.section	.text.startup,"ax",@progbits
