DIR 3927f807e20ac446f697d28db8904cc3
--- pr28982a.c.{atmega128}.{2}.{vanilla}.s	2020-08-05 18:46:45.173432565 +0000
+++ pr28982a.c.{atmega128}.{2}.{ccmode}.s	2020-08-05 18:46:44.345426922 +0000
@@ -42,22 +42,22 @@
 /* stack size = 170 */
 .L__stack_usage = 170
 	movw r18,r24
-	lds r26,incs
-	lds r27,incs+1
-	lds r30,incs+2
-	lds r31,incs+2+1
-	lds r22,incs+4
-	lds r23,incs+4+1
-	lds r20,incs+6
-	lds r21,incs+6+1
-	lds r24,incs+8
-	lds r25,incs+8+1
-	lds r16,incs+10
-	lds r17,incs+10+1
+	lds r16,incs
+	lds r17,incs+1
 	adiw r28,109-62
 	std Y+63,r17
 	std Y+62,r16
 	sbiw r28,109-62
+	lds r26,incs+2
+	lds r27,incs+2+1
+	lds r30,incs+4
+	lds r31,incs+4+1
+	lds r22,incs+6
+	lds r23,incs+6+1
+	lds r20,incs+8
+	lds r21,incs+8+1
+	lds r24,incs+10
+	lds r25,incs+10+1
 	lds r16,incs+12
 	lds r17,incs+12+1
 	adiw r28,111-62
@@ -232,6 +232,18 @@
 	or r18,r19
 	brne .+2
 	rjmp .L4
+	adiw r28,109-62
+	ldd r16,Y+62
+	ldd r17,Y+63
+	sbiw r28,109-62
+	lsl r16
+	rol r17
+	lsl r16
+	rol r17
+	adiw r28,109-62
+	std Y+63,r17
+	std Y+62,r16
+	sbiw r28,109-62
 	movw r16,r26
 	lsl r16
 	rol r17
@@ -287,18 +299,6 @@
 	st Y,r16
 	subi r28,lo8(147)
 	sbci r29,hi8(147)
-	adiw r28,109-62
-	ldd r16,Y+62
-	ldd r17,Y+63
-	sbiw r28,109-62
-	lsl r16
-	rol r17
-	lsl r16
-	rol r17
-	adiw r28,109-62
-	std Y+63,r17
-	std Y+62,r16
-	sbiw r28,109-62
 	adiw r28,111-62
 	ldd r16,Y+62
 	ldd r17,Y+63
@@ -698,12 +698,10 @@
 	std Y+30,r23
 	std Y+31,r24
 	std Y+32,r25
-	subi r28,lo8(-139)
-	sbci r29,hi8(-139)
-	ld r18,Y
-	ldd r19,Y+1
-	subi r28,lo8(139)
-	sbci r29,hi8(139)
+	adiw r28,109-62
+	ldd r18,Y+62
+	ldd r19,Y+63
+	sbiw r28,109-62
 	add r16,r18
 	adc r17,r19
 	movw r30,r14
@@ -728,12 +726,12 @@
 	std Y+34,r23
 	std Y+35,r24
 	std Y+36,r25
-	subi r28,lo8(-141)
-	sbci r29,hi8(-141)
+	subi r28,lo8(-139)
+	sbci r29,hi8(-139)
 	ld r18,Y
 	ldd r19,Y+1
-	subi r28,lo8(141)
-	sbci r29,hi8(141)
+	subi r28,lo8(139)
+	sbci r29,hi8(139)
 	add r14,r18
 	adc r15,r19
 	movw r30,r12
@@ -758,12 +756,12 @@
 	std Y+38,r23
 	std Y+39,r24
 	std Y+40,r25
-	subi r28,lo8(-143)
-	sbci r29,hi8(-143)
+	subi r28,lo8(-141)
+	sbci r29,hi8(-141)
 	ld r18,Y
 	ldd r19,Y+1
-	subi r28,lo8(143)
-	sbci r29,hi8(143)
+	subi r28,lo8(141)
+	sbci r29,hi8(141)
 	add r12,r18
 	adc r13,r19
 	movw r30,r10
@@ -788,12 +786,12 @@
 	std Y+42,r23
 	std Y+43,r24
 	std Y+44,r25
-	subi r28,lo8(-145)
-	sbci r29,hi8(-145)
+	subi r28,lo8(-143)
+	sbci r29,hi8(-143)
 	ld r18,Y
 	ldd r19,Y+1
-	subi r28,lo8(145)
-	sbci r29,hi8(145)
+	subi r28,lo8(143)
+	sbci r29,hi8(143)
 	add r10,r18
 	adc r11,r19
 	movw r30,r8
@@ -818,12 +816,12 @@
 	std Y+46,r23
 	std Y+47,r24
 	std Y+48,r25
-	subi r28,lo8(-147)
-	sbci r29,hi8(-147)
+	subi r28,lo8(-145)
+	sbci r29,hi8(-145)
 	ld r18,Y
 	ldd r19,Y+1
-	subi r28,lo8(147)
-	sbci r29,hi8(147)
+	subi r28,lo8(145)
+	sbci r29,hi8(145)
 	add r8,r18
 	adc r9,r19
 	movw r30,r6
@@ -848,10 +846,12 @@
 	std Y+50,r23
 	std Y+51,r24
 	std Y+52,r25
-	adiw r28,109-62
-	ldd r18,Y+62
-	ldd r19,Y+63
-	sbiw r28,109-62
+	subi r28,lo8(-147)
+	sbci r29,hi8(-147)
+	ld r18,Y
+	ldd r19,Y+1
+	subi r28,lo8(147)
+	sbci r29,hi8(147)
 	add r6,r18
 	adc r7,r19
 	movw r30,r4
