--- 931004-2.c.{atmega128}.{s}.{vanilla}.s	2020-08-05 15:53:13.557098052 +0000
+++ 931004-2.c.{atmega128}.{s}.{ccmode}.s	2020-08-05 15:53:13.325095733 +0000
@@ -19,13 +19,13 @@
 .L__stack_usage = 2
 	movw r30,r28
 	adiw r30,5
-	ld r20,Z+
-	ld r21,Z+
+	ld r18,Z+
+	ld r19,Z+
 	ldi r25,0
 	ldi r24,0
 .L2:
-	cp r24,r20
-	cpc r25,r21
+	cp r24,r18
+	cpc r25,r19
 	brlt .L4
 	ld r24,Z
 	ldd r25,Z+1
@@ -41,13 +41,13 @@
 	pop r28
 	ret
 .L4:
+	movw r20,r24
+	subi r20,-10
+	sbci r21,-1
 	ld r22,Z+
 	ld r23,Z+
-	movw r18,r24
-	subi r18,-10
-	sbci r19,-1
-	cp r22,r18
-	cpc r23,r19
+	cp r22,r20
+	cpc r23,r21
 	breq .L3
 .L6:
 	call abort
