DIR 2a2c10fb0ef37bc574f4773fcc363e7f
--- ashldi-1.c.{atmega128}.{3}.{vanilla}.s	2020-08-05 18:45:53.125078008 +0000
+++ ashldi-1.c.{atmega128}.{3}.{ccmode}.s	2020-08-05 18:45:52.157071418 +0000
@@ -63,10 +63,11 @@
 	cpi r30,1
 	cpc r31,__zero_reg__
 	breq .L4
+.L12:
 	movw r24,r30
 	sbiw r24,2
 	sbiw r24,62
-	brsh .L14
+	brsh .L15
 .L5:
 	adiw r30,1
 	adiw r26,8
@@ -76,7 +77,7 @@
 	ldi r25,0
 	ldi r24,0
 	call exit
-.L14:
+.L15:
 	ld r18,X
 	adiw r26,1
 	ld r19,X
@@ -145,7 +146,7 @@
 	adiw r26,8
 	ldi r30,lo8(2)
 	ldi r31,0
-	rjmp .L8
+	rjmp .L12
 	.size	main, .-main
 	.section	.rodata
 	.type	data, @object
