DIR 22bb9dd13d2e2afb9fab90a12c95d1be
--- pr85582-1.c.{avr25}.{s}.{vanilla}.s	2020-08-14 11:38:47.780070569 +0000
+++ pr85582-1.c.{avr25}.{s}.{ccmode}.s	2020-08-14 11:38:50.144041778 +0000
@@ -32,14 +32,14 @@
 	lds r15,c+5
 	lds r30,c+6
 	lds r17,c+7
+	movw r26,r8
+	andi r26,5
+	clr r27
 	movw r18,r10
 	movw r20,r12
 	movw r22,r14
 	mov r24,r30
 	mov r25,r17
-	movw r26,r8
-	andi r26,5
-	clr r27
 	mov r16,r26
 	rcall __ashrdi3
 	movw r6,r18
