DIR 478a675a90ef88e1746ba1816bf9b332
--- cmpdi-1.c.{atmega128}.{2}.{vanilla}.s	2020-08-05 18:47:35.294111232 +0000
+++ cmpdi-1.c.{atmega128}.{2}.{ccmode}.s	2020-08-05 18:47:34.610099624 +0000
@@ -430,8 +430,8 @@
 	std Y+13,r18
 	rjmp .L45
 .L41:
-	ldi r24,13
-	cp r6,r24
+	ldi r30,13
+	cp r6,r30
 	cpc r7,__zero_reg__
 	breq .+2
 	rjmp .L36
@@ -466,8 +466,8 @@
 	call __cmpdi2
 	brlo .+2
 	rjmp .L63
-	ldi r18,13
-	cp r6,r18
+	ldi r24,13
+	cp r6,r24
 	cpc r7,__zero_reg__
 	breq .+2
 	rjmp .L36
@@ -627,8 +627,8 @@
 	breq .+4
 	brlo .+2
 	rjmp .L41
-	ldi r19,-116
-	cp r6,r19
+	ldi r25,-116
+	cp r6,r25
 	cpc r7,__zero_reg__
 	brne .L36
 	ldi r24,lo8(13)
@@ -695,8 +695,8 @@
 .L36:
 	call abort
 .L63:
-	ldi r31,-116
-	cp r6,r31
+	ldi r19,-116
+	cp r6,r19
 	cpc r7,__zero_reg__
 	brne .L36
 	ldi r24,lo8(13)
@@ -709,8 +709,8 @@
 	std Y+9,r30
 	ldd r30,Y+11
 	ldd r31,Y+12
-	cp r26,r30
-	cpc r27,r31
+	subi r30,lo8(args+64)
+	sbci r31,hi8(args+64)
 	breq .+2
 	rjmp .L46
 	ldi r25,0
