DIR bc39ad903bf76a54cbf77eff0b13646d
--- pr51856.c.{atmega128}.{s}.{vanilla}.s	2020-08-07 14:47:18.002499346 +0000
+++ pr51856.c.{atmega128}.{s}.{ccmode}.s	2020-08-07 14:47:18.262501491 +0000
@@ -70,6 +70,8 @@
 	movw r24,r28
 	adiw r24,1
 	call fn2
+	ldi r21,0
+	ldi r20,0
 	ldi r24,lo8(16)
 	mov r14,r24
 	mov r15,__zero_reg__
@@ -78,8 +80,6 @@
 	movw r18,r28
 	subi r18,-1
 	sbci r19,-3
-	ldi r21,0
-	ldi r20,0
 	ldi r22,lo8(.LC0)
 	ldi r23,hi8(.LC0)
 	lds r24,r
