DIR 9dae99b6d127c98ff333a76a4a559467
--- 20010915-1.c.{avr25}.{s}.{vanilla}.s	2020-08-10 18:58:40.244626538 +0000
+++ 20010915-1.c.{avr25}.{s}.{ccmode}.s	2020-08-10 18:58:46.124652916 +0000
@@ -99,10 +99,6 @@
 .global	x
 	.type	x, @function
 x:
-	push r8
-	push r9
-	push r10
-	push r11
 	push r12
 	push r13
 	push r14
@@ -116,8 +112,8 @@
 	in r29,__SP_H__
 /* prologue: function */
 /* frame size = 2 */
-/* stack size = 14 */
-.L__stack_usage = 14
+/* stack size = 10 */
+.L__stack_usage = 10
 	movw r14,r24
 	movw r16,r22
 	std Y+2,__zero_reg__
@@ -140,18 +136,18 @@
 	ldd r25,Z+1
 	sbiw r24,0
 	breq .L9
-	movw r30,r28
-	adiw r30,1
-	movw r10,r30
-	movw r22,r30
+	movw r22,r28
+	subi r22,-1
+	sbci r23,-1
 	rcall s
-	movw r8,r10
 	sbiw r24,0
 	breq .L10
 	movw r30,r24
 	st Z+,__zero_reg__
 	movw r12,r30
-	movw r22,r10
+	movw r22,r28
+	subi r22,-1
+	sbci r23,-1
 	movw r24,r30
 	rcall s
 	sts h+1,r25
@@ -170,7 +166,9 @@
 	rol r31
 	add r30,r16
 	adc r31,r17
-	movw r22,r8
+	movw r22,r28
+	subi r22,-1
+	sbci r23,-1
 	ld r24,Z
 	ldd r25,Z+1
 	rcall s
@@ -212,10 +210,6 @@
 	pop r14
 	pop r13
 	pop r12
-	pop r11
-	pop r10
-	pop r9
-	pop r8
 	ret
 .L13:
 	lsl r24
