DIR 8700a7df3bad1a367d0aefac578e1346
--- 20041214-1.c.{atmega128}.{3}.{vanilla}.s	2020-08-05 18:50:50.637131504 +0000
+++ 20041214-1.c.{atmega128}.{3}.{ccmode}.s	2020-08-05 18:50:50.449128845 +0000
@@ -23,8 +23,8 @@
 	movw r16,r22
 	movw r30,r22
 	ld r24,Z
-	cpse r24,__zero_reg__
-	rjmp .L3
+	tst r24
+	brne .L3
 	rjmp .L2
 .L6:
 	movw r20,r28
@@ -123,13 +123,18 @@
 	ldd r25,Y+2
 	cpi r24,97
 	sbci r25,115
-	breq .L16
-.L13:
-	ldi r24,lo8(1)
-	ldi r25,0
-.L14:
-	or r24,r25
-	brne .L17
+	breq .L18
+.L16:
+	call abort
+.L18:
+	ldd r24,Y+3
+	ldd r25,Y+4
+	cpi r24,100
+	sbci r25,102
+	brne .L16
+	ldd r24,Y+5
+	cpse r24,__zero_reg__
+	rjmp .L16
 	ldi r25,0
 	ldi r24,0
 /* epilogue start */
@@ -140,20 +145,6 @@
 	out __SREG__,__tmp_reg__
 	out __SP_L__,r28
 	ret
-.L16:
-	ldd r24,Y+3
-	ldd r25,Y+4
-	cpi r24,100
-	sbci r25,102
-	brne .L13
-	ldd r24,Y+5
-	cpse r24,__zero_reg__
-	rjmp .L13
-	ldi r25,0
-	ldi r24,0
-	rjmp .L14
-.L17:
-	call abort
 	.size	main, .-main
-	.ident	"GCC: (GNU) 11.0.0 20200802 (experimental)"
+	.ident	"GCC: (GNU) 11.0.0 20200804 (experimental)"
 .global __do_copy_data
