DIR 1dc20e7e86d3e810416ce126d4bc6036
--- 20000706-1.c.{atmega128}.{s}.{vanilla}.s	2020-08-13 16:47:03.180017433 +0000
+++ 20000706-1.c.{atmega128}.{s}.{ccmode}.s	2020-08-13 16:47:04.847994058 +0000
@@ -146,21 +146,17 @@
 /* frame size = 10 */
 /* stack size = 10 */
 .L__stack_usage = 10
-	ldi r24,lo8(1)
-	mov r16,r24
-	ldi r24,lo8(2)
-	mov r18,r24
-	ldi r24,lo8(3)
-	mov r20,r24
-	ldi r24,lo8(4)
-	mov r22,r24
-	ldi r24,lo8(5)
 	mov r15,__zero_reg__
 	mov r14,__zero_reg__
+	ldi r16,lo8(1)
 	ldi r17,0
+	ldi r18,lo8(2)
 	ldi r19,0
+	ldi r20,lo8(3)
 	ldi r21,0
+	ldi r22,lo8(4)
 	ldi r23,0
+	ldi r24,lo8(5)
 	ldi r25,0
 	call foo
 	ldi r25,0
