DIR ad9f915e6ff3fc589593fc864f58ef87
--- pr53645.c.{avr3}.{2}.{vanilla}.s	2020-08-10 19:06:23.094129715 +0000
+++ pr53645.c.{avr3}.{2}.{ccmode}.s	2020-08-10 19:07:12.118239788 +0000
@@ -3311,16 +3311,16 @@
 main:
 	in r28,__SP_L__
 	in r29,__SP_H__
-	sbiw r28,51
+	sbiw r28,49
 	in __tmp_reg__,__SREG__
 	cli
 	out __SP_H__,r29
 	out __SREG__,__tmp_reg__
 	out __SP_L__,r28
 /* prologue: function */
-/* frame size = 51 */
-/* stack size = 51 */
-.L__stack_usage = 51
+/* frame size = 49 */
+/* stack size = 49 */
+.L__stack_usage = 49
 	mov r17,r29
 	mov r16,r28
 	subi r16,-16
@@ -3330,13 +3330,13 @@
 	mov r12,r27
 	ldi r27,hi8(u)
 	mov r13,r27
-	std Y+51,__zero_reg__
-	std Y+50,__zero_reg__
-	mov r25,r17
-	mov r24,r16
-	adiw r24,16
-	mov r15,r25
-	mov r14,r24
+	mov r3,__zero_reg__
+	mov r2,__zero_reg__
+	mov r15,r17
+	mov r14,r16
+	ldi r18,16
+	add r14,r18
+	adc r15,__zero_reg__
 	ldi r24,lo8(3)
 	mov r8,r24
 	mov r9,__zero_reg__
@@ -3355,23 +3355,22 @@
 	mov r25,r15
 	mov r24,r14
 	call uq4444
-	ldd r26,Y+50
-	ldd r27,Y+51
-	swap r26
-	swap r27
-	andi r27,0xf0
-	eor r27,r26
-	andi r26,0xf0
-	eor r27,r26
-	std Y+49,r27
-	std Y+48,r26
+	mov r25,r3
+	mov r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	mov r27,r25
+	mov r26,r24
 	subi r26,lo8(-(u))
 	sbci r27,hi8(-(u))
-	mov r3,r27
-	mov r2,r26
+	std Y+49,r27
+	std Y+48,r26
 	ld r24,X+
 	ld r25,X
-	sbiw r26,1
 	lsr r25
 	ror r24
 	lsr r25
@@ -3384,7 +3383,8 @@
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	mov r26,r2
+	ldd r26,Y+48
+	ldd r27,Y+49
 	adiw r26,6
 	ld r24,X+
 	ld r25,X
@@ -3403,26 +3403,31 @@
 	adiw r26,4
 	ld r24,X+
 	ld r25,X
-	sbiw r26,4+1
 	lsr r25
 	ror r24
 	lsr r25
 	ror r24
-	ldd r18,Z+20
-	ldd r19,Z+21
+	mov r27,r17
+	mov r26,r16
+	adiw r26,20
+	ld r18,X+
+	ld r19,X
+	sbiw r26,20+1
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	adiw r26,2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ldd r24,Z+2
+	ldd r25,Z+3
 	lsr r25
 	ror r24
 	lsr r25
 	ror r24
-	ldd r18,Z+18
-	ldd r19,Z+19
+	adiw r26,18
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
@@ -3432,8 +3437,8 @@
 	mov r25,r15
 	mov r24,r14
 	call ur4444
-	mov r27,r3
-	mov r26,r2
+	ldd r26,Y+48
+	ldd r27,Y+49
 	ld r24,X+
 	ld r25,X
 	sbiw r26,1
@@ -3462,22 +3467,27 @@
 	adiw r26,4
 	ld r24,X+
 	ld r25,X
-	sbiw r26,4+1
 	andi r24,3
 	clr r25
-	ldd r18,Z+20
-	ldd r19,Z+21
+	mov r27,r17
+	mov r26,r16
+	adiw r26,20
+	ld r18,X+
+	ld r19,X
+	sbiw r26,20+1
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	adiw r26,2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ldd r24,Z+2
+	ldd r25,Z+3
 	andi r24,3
 	clr r25
-	ldd r18,Z+18
-	ldd r19,Z+19
+	adiw r26,18
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
@@ -3493,8 +3503,8 @@
 	ld r18,X+
 	ld r19,X
 	sbiw r26,16+1
-	mov r31,r3
-	mov r30,r2
+	ldd r30,Y+48
+	ldd r31,Y+49
 	ld r24,Z
 	ldd r25,Z+1
 	cp r18,r24
@@ -3516,14 +3526,22 @@
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Y+48
-	ldd r25,Y+49
-	subi r24,lo8(-(u))
-	sbci r25,hi8(-(u))
-	mov r3,r25
-	mov r2,r24
-	mov r27,r25
-	mov r26,r24
+	mov r25,r3
+	mov r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	mov r19,r25
+	mov r18,r24
+	subi r18,lo8(-(u))
+	sbci r19,hi8(-(u))
+	std Y+49,r19
+	std Y+48,r18
+	mov r27,r19
+	mov r26,r18
 	adiw r26,4
 	ld r24,X+
 	ld r25,X
@@ -3565,8 +3583,8 @@
 	or r24,r25
 	breq .+2
 	rjmp .L42
-	mov r31,r3
-	mov r30,r2
+	ldd r30,Y+48
+	ldd r31,Y+49
 	ldd r24,Z+6
 	ldd r25,Z+7
 	andi r24,7
@@ -3574,33 +3592,30 @@
 	adiw r26,22
 	ld r18,X+
 	ld r19,X
+	sbiw r26,22+1
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
-	adiw r26,4
-	ld r24,X+
-	ld r25,X
-	sbiw r26,4+1
+	ldd r24,Z+4
+	ldd r25,Z+5
 	andi r24,1
 	clr r25
-	mov r31,r17
-	mov r30,r16
-	ldd r18,Z+20
-	ldd r19,Z+21
+	adiw r26,20
+	ld r18,X+
+	ld r19,X
+	sbiw r26,20+1
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	adiw r26,2
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+2
+	ldd r25,Z+3
 	andi r24,3
 	clr r25
-	ldd r18,Z+18
-	ldd r19,Z+19
+	adiw r26,18
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
@@ -3610,58 +3625,57 @@
 	mov r25,r15
 	mov r24,r14
 	call uq3333
-	mov r27,r3
-	mov r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	mov r23,r9
 	mov r22,r8
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
-	ldd r24,Z+16
-	ldd r25,Z+17
+	mov r27,r17
+	mov r26,r16
+	adiw r26,16
+	ld r24,X+
+	ld r25,X
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	mov r23,r9
 	mov r22,r8
 	call __udivmodhi4
+	mov r31,r17
+	mov r30,r16
 	ldd r24,Z+22
 	ldd r25,Z+23
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r31,r3
-	mov r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	mov r23,r9
 	mov r22,r8
 	call __udivmodhi4
-	mov r27,r17
-	mov r26,r16
-	adiw r26,20
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+20
+	ldd r25,Z+21
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	mov r23,r9
 	mov r22,r8
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
 	ldd r24,Z+18
 	ldd r25,Z+19
 	cp r24,r22
@@ -3673,64 +3687,69 @@
 	mov r25,r15
 	mov r24,r14
 	call ur3333
-	ldd r24,Y+48
-	ldd r25,Y+49
-	subi r24,lo8(-(u))
-	sbci r25,hi8(-(u))
-	mov r3,r25
-	mov r2,r24
-	mov r27,r25
-	mov r26,r24
-	ld r24,X+
-	ld r25,X
+	mov r25,r3
+	mov r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	mov r31,r25
+	mov r30,r24
+	subi r30,lo8(-(u))
+	sbci r31,hi8(-(u))
+	std Y+49,r31
+	std Y+48,r30
+	ld r24,Z
+	ldd r25,Z+1
 	mov r23,r9
 	mov r22,r8
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
-	ldd r18,Z+16
-	ldd r19,Z+17
+	mov r27,r17
+	mov r26,r16
+	adiw r26,16
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	mov r23,r9
 	mov r22,r8
 	call __udivmodhi4
+	mov r31,r17
+	mov r30,r16
 	ldd r18,Z+22
 	ldd r19,Z+23
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	mov r31,r3
-	mov r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	mov r23,r9
 	mov r22,r8
 	call __udivmodhi4
-	mov r27,r17
-	mov r26,r16
-	adiw r26,20
-	ld r18,X+
-	ld r19,X
+	ldd r18,Z+20
+	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	mov r23,r9
 	mov r22,r8
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
 	ldd r18,Z+18
 	ldd r19,Z+19
 	cp r18,r24
@@ -3742,58 +3761,57 @@
 	mov r25,r15
 	mov r24,r14
 	call uq6565
-	mov r27,r3
-	mov r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	mov r23,r7
 	mov r22,r6
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
-	ldd r24,Z+16
-	ldd r25,Z+17
+	mov r27,r17
+	mov r26,r16
+	adiw r26,16
+	ld r24,X+
+	ld r25,X
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	ldi r22,lo8(5)
 	ldi r23,0
 	call __udivmodhi4
+	mov r31,r17
+	mov r30,r16
 	ldd r24,Z+22
 	ldd r25,Z+23
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r31,r3
-	mov r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	mov r23,r7
 	mov r22,r6
 	call __udivmodhi4
-	mov r27,r17
-	mov r26,r16
-	adiw r26,20
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+20
+	ldd r25,Z+21
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	ldi r22,lo8(5)
 	ldi r23,0
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
 	ldd r24,Z+18
 	ldd r25,Z+19
 	cp r24,r22
@@ -3805,62 +3823,71 @@
 	mov r25,r15
 	mov r24,r14
 	call ur6565
-	mov r27,r3
-	mov r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	mov r23,r7
 	mov r22,r6
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
-	ldd r18,Z+16
-	ldd r19,Z+17
+	mov r27,r17
+	mov r26,r16
+	adiw r26,16
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	ldi r22,lo8(5)
 	ldi r23,0
 	call __udivmodhi4
+	mov r31,r17
+	mov r30,r16
 	ldd r18,Z+22
 	ldd r19,Z+23
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r30,Y+48
-	ldd r31,Y+49
-	subi r30,lo8(-(u))
-	sbci r31,hi8(-(u))
-	mov r3,r31
-	mov r2,r30
-	ldd r24,Z+4
-	ldd r25,Z+5
+	mov r25,r3
+	mov r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	mov r19,r25
+	mov r18,r24
+	subi r18,lo8(-(u))
+	sbci r19,hi8(-(u))
+	std Y+49,r19
+	std Y+48,r18
+	mov r27,r19
+	mov r26,r18
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	mov r23,r7
 	mov r22,r6
 	call __udivmodhi4
-	mov r27,r17
-	mov r26,r16
-	adiw r26,20
-	ld r18,X+
-	ld r19,X
+	ldd r18,Z+20
+	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	ldi r22,lo8(5)
 	ldi r23,0
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
 	ldd r18,Z+18
 	ldd r19,Z+19
 	cp r18,r24
@@ -3872,58 +3899,57 @@
 	mov r25,r15
 	mov r24,r14
 	call uq1414146
-	mov r27,r3
-	mov r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	mov r23,r5
 	mov r22,r4
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
-	ldd r24,Z+16
-	ldd r25,Z+17
+	mov r27,r17
+	mov r26,r16
+	adiw r26,16
+	ld r24,X+
+	ld r25,X
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	mov r23,r7
 	mov r22,r6
 	call __udivmodhi4
+	mov r31,r17
+	mov r30,r16
 	ldd r24,Z+22
 	ldd r25,Z+23
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r31,r3
-	mov r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	mov r23,r5
 	mov r22,r4
 	call __udivmodhi4
-	mov r27,r17
-	mov r26,r16
-	adiw r26,20
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+20
+	ldd r25,Z+21
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	mov r23,r5
 	mov r22,r4
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
 	ldd r24,Z+18
 	ldd r25,Z+19
 	cp r24,r22
@@ -3935,58 +3961,57 @@
 	mov r25,r15
 	mov r24,r14
 	call ur1414146
-	mov r27,r3
-	mov r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	mov r23,r5
 	mov r22,r4
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
-	ldd r18,Z+16
-	ldd r19,Z+17
+	mov r27,r17
+	mov r26,r16
+	adiw r26,16
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	mov r23,r7
 	mov r22,r6
 	call __udivmodhi4
+	mov r31,r17
+	mov r30,r16
 	ldd r18,Z+22
 	ldd r19,Z+23
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	mov r31,r3
-	mov r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	mov r23,r5
 	mov r22,r4
 	call __udivmodhi4
-	mov r27,r17
-	mov r26,r16
-	adiw r26,20
-	ld r18,X+
-	ld r19,X
+	ldd r18,Z+20
+	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	mov r23,r5
 	mov r22,r4
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
 	ldd r18,Z+18
 	ldd r19,Z+19
 	cp r18,r24
@@ -3998,64 +4023,69 @@
 	mov r25,r15
 	mov r24,r14
 	call uq7777
-	ldd r24,Y+48
-	ldd r25,Y+49
-	subi r24,lo8(-(u))
-	sbci r25,hi8(-(u))
-	mov r3,r25
-	mov r2,r24
-	mov r27,r25
-	mov r26,r24
-	ld r24,X+
-	ld r25,X
+	mov r25,r3
+	mov r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	mov r31,r25
+	mov r30,r24
+	subi r30,lo8(-(u))
+	sbci r31,hi8(-(u))
+	std Y+49,r31
+	std Y+48,r30
+	ld r24,Z
+	ldd r25,Z+1
 	mov r23,r11
 	mov r22,r10
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
-	ldd r24,Z+16
-	ldd r25,Z+17
+	mov r27,r17
+	mov r26,r16
+	adiw r26,16
+	ld r24,X+
+	ld r25,X
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	mov r23,r11
 	mov r22,r10
 	call __udivmodhi4
+	mov r31,r17
+	mov r30,r16
 	ldd r24,Z+22
 	ldd r25,Z+23
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r31,r3
-	mov r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	mov r23,r11
 	mov r22,r10
 	call __udivmodhi4
-	mov r27,r17
-	mov r26,r16
-	adiw r26,20
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+20
+	ldd r25,Z+21
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	mov r23,r11
 	mov r22,r10
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
 	ldd r24,Z+18
 	ldd r25,Z+19
 	cp r24,r22
@@ -4067,93 +4097,91 @@
 	mov r25,r15
 	mov r24,r14
 	call ur7777
-	mov r27,r3
-	mov r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	mov r23,r11
 	mov r22,r10
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
-	ldd r18,Z+16
-	ldd r19,Z+17
+	mov r27,r17
+	mov r26,r16
+	adiw r26,16
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	mov r23,r11
 	mov r22,r10
 	call __udivmodhi4
+	mov r31,r17
+	mov r30,r16
 	ldd r18,Z+22
 	ldd r19,Z+23
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	mov r31,r3
-	mov r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	mov r23,r11
 	mov r22,r10
 	call __udivmodhi4
-	mov r27,r17
-	mov r26,r16
-	adiw r26,20
-	ld r18,X+
-	ld r19,X
+	ldd r18,Z+20
+	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	mov r23,r11
 	mov r22,r10
 	call __udivmodhi4
-	mov r31,r17
-	mov r30,r16
 	ldd r18,Z+18
 	ldd r19,Z+19
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldi r31,16
-	add r12,r31
+	ldi r27,16
+	add r12,r27
 	adc r13,__zero_reg__
-	ldd r24,Y+50
-	ldd r25,Y+51
-	sbiw r24,1
+	dec r2
+	or r2,r3
 	breq .L61
-	ldi r25,lo8(1)
-	std Y+50,r25
-	std Y+51,__zero_reg__
+	clr r2
+	inc r2
+	mov r3,__zero_reg__
 	rjmp .L43
 .L61:
-	ldi r26,lo8(s)
-	mov r12,r26
-	ldi r26,hi8(s)
-	mov r13,r26
-	std Y+49,__zero_reg__
-	std Y+48,__zero_reg__
-	ldi r27,lo8(3)
-	mov r8,r27
+	ldi r20,lo8(s)
+	mov r12,r20
+	ldi r20,hi8(s)
+	mov r13,r20
+	mov r15,__zero_reg__
+	mov r14,__zero_reg__
+	ldi r21,lo8(3)
+	mov r8,r21
 	mov r9,__zero_reg__
-	ldi r24,lo8(6)
-	mov r6,r24
+	ldi r22,lo8(6)
+	mov r6,r22
 	mov r7,__zero_reg__
-	ldi r25,lo8(14)
-	mov r4,r25
+	ldi r23,lo8(14)
+	mov r4,r23
 	mov r5,__zero_reg__
-	ldi r18,lo8(7)
-	mov r10,r18
+	ldi r30,lo8(7)
+	mov r10,r30
 	mov r11,__zero_reg__
 .L58:
 	mov r23,r13
@@ -4161,17 +4189,14 @@
 	mov r25,r17
 	mov r24,r16
 	call sq4444
-	ldd r14,Y+48
-	ldd r15,Y+49
-	swap r14
-	swap r15
-	ldi r31,0xf0
-	and r15,r31
-	eor r15,r14
-	and r14,r31
-	eor r15,r14
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ld r24,Z
@@ -4193,6 +4218,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+6
@@ -4214,6 +4245,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+4
@@ -4236,6 +4273,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+2
@@ -4262,6 +4305,12 @@
 	call sr4444
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ld r18,Z
@@ -4287,6 +4336,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+6
@@ -4312,6 +4367,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+4
@@ -4338,6 +4399,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+2
@@ -4368,6 +4435,12 @@
 	call sq1428
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	mov r27,r17
@@ -4401,6 +4474,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+4
@@ -4421,6 +4500,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+2
@@ -4454,6 +4539,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+6
@@ -4479,6 +4570,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+4
@@ -4505,6 +4602,12 @@
 	rjmp .L42
 	mov r31,r15
 	mov r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+2
@@ -4535,12 +4638,20 @@
 	call sq3333
 	mov r25,r15
 	mov r24,r14
-	subi r24,lo8(-(s))
-	sbci r25,hi8(-(s))
-	mov r3,r25
-	mov r2,r24
-	mov r27,r25
-	mov r26,r24
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	mov r19,r25
+	mov r18,r24
+	subi r18,lo8(-(s))
+	sbci r19,hi8(-(s))
+	mov r3,r19
+	mov r2,r18
+	mov r27,r19
+	mov r26,r18
 	ld r24,X+
 	ld r25,X
 	mov r23,r9
@@ -4554,8 +4665,8 @@
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r27,r3
-	mov r26,r2
+	mov r27,r19
+	mov r26,r18
 	adiw r26,6
 	ld r24,X+
 	ld r25,X
@@ -4568,8 +4679,8 @@
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	mov r31,r3
-	mov r30,r2
+	mov r31,r19
+	mov r30,r18
 	ldd r24,Z+4
 	ldd r25,Z+5
 	mov r23,r9
@@ -5032,19 +5143,19 @@
 	ldi r31,16
 	add r12,r31
 	adc r13,__zero_reg__
-	ldd r24,Y+48
-	ldd r25,Y+49
-	sbiw r24,1
+	ldi r18,1
+	cp r14,r18
+	cpc r15,__zero_reg__
 	breq .L62
-	ldi r25,lo8(1)
-	std Y+48,r25
-	std Y+49,__zero_reg__
+	clr r14
+	inc r14
+	mov r15,__zero_reg__
 	rjmp .L58
 .L62:
 	ldi r25,0
 	ldi r24,0
 /* epilogue start */
-	adiw r28,51
+	adiw r28,49
 	in __tmp_reg__,__SREG__
 	cli
 	out __SP_H__,r29
