DIR 00b1f60d196f542c64a7e2f8b1fcac8f
--- 960311-2.c.{atmega128}.{s}.{vanilla}.s	2020-08-06 12:04:58.210318146 +0000
+++ 960311-2.c.{atmega128}.{s}.{lra}.s	2020-08-06 12:04:59.246325395 +0000
@@ -24,29 +24,29 @@
 .global	b
 	.type	b, @function
 b:
-	push r28
-	push r29
+	push r16
+	push r17
 /* prologue: function */
 /* frame size = 0 */
 /* stack size = 2 */
 .L__stack_usage = 2
-	movw r28,r24
+	movw r16,r24
 	sbrc r25,7
 	call a1
 .L3:
-	sbrc r29,6
+	sbrc r17,6
 	call a1
 .L4:
-	sbrs r29,5
+	sbrs r17,5
 	rjmp .L2
 /* epilogue start */
-	pop r29
-	pop r28
+	pop r17
+	pop r16
 	jmp a1
 .L2:
 /* epilogue start */
-	pop r29
-	pop r28
+	pop r17
+	pop r16
 	ret
 	.size	b, .-b
 	.section	.text.startup,"ax",@progbits
