DIR ad9f915e6ff3fc589593fc864f58ef87
--- pr53645.c.{avr35}.{2}.{vanilla}.s	2020-08-10 19:06:17.922117705 +0000
+++ pr53645.c.{avr35}.{2}.{ccmode}.s	2020-08-10 19:06:53.250198211 +0000
@@ -3083,16 +3083,16 @@
 main:
 	in r28,__SP_L__
 	in r29,__SP_H__
-	sbiw r28,51
+	sbiw r28,49
 	in __tmp_reg__,__SREG__
 	cli
 	out __SP_H__,r29
 	out __SREG__,__tmp_reg__
 	out __SP_L__,r28
 /* prologue: function */
-/* frame size = 51 */
-/* stack size = 51 */
-.L__stack_usage = 51
+/* frame size = 49 */
+/* stack size = 49 */
+.L__stack_usage = 49
 	movw r16,r28
 	subi r16,-16
 	sbci r17,-1
@@ -3101,11 +3101,12 @@
 	mov r12,r27
 	ldi r27,hi8(u)
 	mov r13,r27
-	std Y+51,__zero_reg__
-	std Y+50,__zero_reg__
-	movw r24,r16
-	adiw r24,16
-	movw r14,r24
+	mov r3,__zero_reg__
+	mov r2,__zero_reg__
+	movw r14,r16
+	ldi r18,16
+	add r14,r18
+	adc r15,__zero_reg__
 	ldi r24,lo8(3)
 	mov r8,r24
 	mov r9,__zero_reg__
@@ -3122,19 +3123,18 @@
 	movw r22,r12
 	movw r24,r14
 	call uq4444
-	ldd r26,Y+50
-	ldd r27,Y+51
-	swap r26
-	swap r27
-	andi r27,0xf0
-	eor r27,r26
-	andi r26,0xf0
-	eor r27,r26
-	std Y+49,r27
-	std Y+48,r26
+	movw r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	movw r26,r24
 	subi r26,lo8(-(u))
 	sbci r27,hi8(-(u))
-	movw r2,r26
+	std Y+49,r27
+	std Y+48,r26
 	ld r24,X+
 	ld r25,X
 	lsr r25
@@ -3148,7 +3148,8 @@
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	movw r26,r2
+	ldd r26,Y+48
+	ldd r27,Y+49
 	adiw r26,6
 	ld r24,X+
 	ld r25,X
@@ -3167,26 +3168,28 @@
 	adiw r26,4
 	ld r24,X+
 	ld r25,X
-	sbiw r26,4+1
 	lsr r25
 	ror r24
 	lsr r25
 	ror r24
+	movw r26,r16
 	ldd r18,Z+20
 	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	adiw r26,2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ldd r24,Z+2
+	ldd r25,Z+3
 	lsr r25
 	ror r24
 	lsr r25
 	ror r24
-	ldd r18,Z+18
-	ldd r19,Z+19
+	adiw r26,18
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
@@ -3194,7 +3197,8 @@
 	movw r22,r12
 	movw r24,r14
 	call ur4444
-	movw r26,r2
+	ldd r26,Y+48
+	ldd r27,Y+49
 	ld r24,X+
 	ld r25,X
 	sbiw r26,1
@@ -3222,22 +3226,24 @@
 	adiw r26,4
 	ld r24,X+
 	ld r25,X
-	sbiw r26,4+1
 	andi r24,3
 	clr r25
+	movw r26,r16
 	ldd r18,Z+20
 	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	adiw r26,2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ldd r24,Z+2
+	ldd r25,Z+3
 	andi r24,3
 	clr r25
-	ldd r18,Z+18
-	ldd r19,Z+19
+	adiw r26,18
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
@@ -3250,7 +3256,8 @@
 	ld r18,X+
 	ld r19,X
 	sbiw r26,16+1
-	movw r30,r2
+	ldd r30,Y+48
+	ldd r31,Y+49
 	ld r24,Z
 	ldd r25,Z+1
 	cp r18,r24
@@ -3272,12 +3279,19 @@
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Y+48
-	ldd r25,Y+49
-	subi r24,lo8(-(u))
-	sbci r25,hi8(-(u))
-	movw r2,r24
-	movw r26,r24
+	movw r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	movw r18,r24
+	subi r18,lo8(-(u))
+	sbci r19,hi8(-(u))
+	std Y+49,r19
+	std Y+48,r18
+	movw r26,r18
 	adiw r26,4
 	ld r24,X+
 	ld r25,X
@@ -3315,7 +3329,8 @@
 	or r24,r25
 	breq .+2
 	rjmp .L42
-	movw r30,r2
+	ldd r30,Y+48
+	ldd r31,Y+49
 	ldd r24,Z+6
 	ldd r25,Z+7
 	andi r24,7
@@ -3323,29 +3338,30 @@
 	adiw r26,22
 	ld r18,X+
 	ld r19,X
+	sbiw r26,22+1
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	movw r26,r2
 	ldd r24,Z+4
 	ldd r25,Z+5
 	andi r24,1
 	clr r25
-	movw r30,r16
-	ldd r18,Z+20
-	ldd r19,Z+21
+	adiw r26,20
+	ld r18,X+
+	ld r19,X
+	sbiw r26,20+1
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	adiw r26,2
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+2
+	ldd r25,Z+3
 	andi r24,3
 	clr r25
-	ldd r18,Z+18
-	ldd r19,Z+19
+	adiw r26,18
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
@@ -3353,48 +3369,51 @@
 	movw r22,r12
 	movw r24,r14
 	call uq3333
-	movw r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	movw r22,r8
 	call __udivmodhi4
-	movw r30,r16
-	ldd r24,Z+16
-	ldd r25,Z+17
+	movw r26,r16
+	adiw r26,16
+	ld r24,X+
+	ld r25,X
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	movw r22,r8
 	call __udivmodhi4
+	movw r30,r16
 	ldd r24,Z+22
 	ldd r25,Z+23
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
-	movw r22,r8
-	call __udivmodhi4
-	movw r26,r16
-	adiw r26,20
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
 	ld r24,X+
 	ld r25,X
+	movw r22,r8
+	call __udivmodhi4
+	ldd r24,Z+20
+	ldd r25,Z+21
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	movw r22,r8
 	call __udivmodhi4
-	movw r30,r16
 	ldd r24,Z+18
 	ldd r25,Z+19
 	cp r24,r22
@@ -3404,53 +3423,61 @@
 	movw r22,r12
 	movw r24,r14
 	call ur3333
-	ldd r24,Y+48
-	ldd r25,Y+49
-	subi r24,lo8(-(u))
-	sbci r25,hi8(-(u))
-	movw r2,r24
-	movw r26,r24
-	ld r24,X+
-	ld r25,X
+	movw r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	movw r30,r24
+	subi r30,lo8(-(u))
+	sbci r31,hi8(-(u))
+	std Y+49,r31
+	std Y+48,r30
+	ld r24,Z
+	ldd r25,Z+1
 	movw r22,r8
 	call __udivmodhi4
-	movw r30,r16
-	ldd r18,Z+16
-	ldd r19,Z+17
+	movw r26,r16
+	adiw r26,16
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	movw r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	movw r22,r8
 	call __udivmodhi4
+	movw r30,r16
 	ldd r18,Z+22
 	ldd r19,Z+23
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	movw r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	movw r22,r8
 	call __udivmodhi4
-	movw r26,r16
-	adiw r26,20
-	ld r18,X+
-	ld r19,X
+	ldd r18,Z+20
+	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	movw r22,r8
 	call __udivmodhi4
-	movw r30,r16
 	ldd r18,Z+18
 	ldd r19,Z+19
 	cp r18,r24
@@ -3460,50 +3487,53 @@
 	movw r22,r12
 	movw r24,r14
 	call uq6565
-	movw r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	movw r22,r6
 	call __udivmodhi4
-	movw r30,r16
-	ldd r24,Z+16
-	ldd r25,Z+17
+	movw r26,r16
+	adiw r26,16
+	ld r24,X+
+	ld r25,X
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	ldi r22,lo8(5)
 	ldi r23,0
 	call __udivmodhi4
+	movw r30,r16
 	ldd r24,Z+22
 	ldd r25,Z+23
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
-	movw r22,r6
-	call __udivmodhi4
-	movw r26,r16
-	adiw r26,20
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
 	ld r24,X+
 	ld r25,X
+	movw r22,r6
+	call __udivmodhi4
+	ldd r24,Z+20
+	ldd r25,Z+21
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	ldi r22,lo8(5)
 	ldi r23,0
 	call __udivmodhi4
-	movw r30,r16
 	ldd r24,Z+18
 	ldd r25,Z+19
 	cp r24,r22
@@ -3513,54 +3543,64 @@
 	movw r22,r12
 	movw r24,r14
 	call ur6565
-	movw r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	movw r22,r6
 	call __udivmodhi4
-	movw r30,r16
-	ldd r18,Z+16
-	ldd r19,Z+17
+	movw r26,r16
+	adiw r26,16
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	movw r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	ldi r22,lo8(5)
 	ldi r23,0
 	call __udivmodhi4
+	movw r30,r16
 	ldd r18,Z+22
 	ldd r19,Z+23
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r30,Y+48
-	ldd r31,Y+49
-	subi r30,lo8(-(u))
-	sbci r31,hi8(-(u))
-	movw r2,r30
-	ldd r24,Z+4
-	ldd r25,Z+5
+	movw r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	movw r18,r24
+	subi r18,lo8(-(u))
+	sbci r19,hi8(-(u))
+	std Y+49,r19
+	std Y+48,r18
+	movw r26,r18
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	movw r22,r6
 	call __udivmodhi4
-	movw r26,r16
-	adiw r26,20
-	ld r18,X+
-	ld r19,X
+	ldd r18,Z+20
+	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	ldi r22,lo8(5)
 	ldi r23,0
 	call __udivmodhi4
-	movw r30,r16
 	ldd r18,Z+18
 	ldd r19,Z+19
 	cp r18,r24
@@ -3570,48 +3610,51 @@
 	movw r22,r12
 	movw r24,r14
 	call uq1414146
-	movw r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	movw r22,r4
 	call __udivmodhi4
-	movw r30,r16
-	ldd r24,Z+16
-	ldd r25,Z+17
+	movw r26,r16
+	adiw r26,16
+	ld r24,X+
+	ld r25,X
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	movw r22,r6
 	call __udivmodhi4
+	movw r30,r16
 	ldd r24,Z+22
 	ldd r25,Z+23
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
-	movw r22,r4
-	call __udivmodhi4
-	movw r26,r16
-	adiw r26,20
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
 	ld r24,X+
 	ld r25,X
+	movw r22,r4
+	call __udivmodhi4
+	ldd r24,Z+20
+	ldd r25,Z+21
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	movw r22,r4
 	call __udivmodhi4
-	movw r30,r16
 	ldd r24,Z+18
 	ldd r25,Z+19
 	cp r24,r22
@@ -3621,48 +3664,51 @@
 	movw r22,r12
 	movw r24,r14
 	call ur1414146
-	movw r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	movw r22,r4
 	call __udivmodhi4
-	movw r30,r16
-	ldd r18,Z+16
-	ldd r19,Z+17
+	movw r26,r16
+	adiw r26,16
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	movw r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	movw r22,r6
 	call __udivmodhi4
+	movw r30,r16
 	ldd r18,Z+22
 	ldd r19,Z+23
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	movw r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	movw r22,r4
 	call __udivmodhi4
-	movw r26,r16
-	adiw r26,20
-	ld r18,X+
-	ld r19,X
+	ldd r18,Z+20
+	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	movw r22,r4
 	call __udivmodhi4
-	movw r30,r16
 	ldd r18,Z+18
 	ldd r19,Z+19
 	cp r18,r24
@@ -3672,53 +3718,61 @@
 	movw r22,r12
 	movw r24,r14
 	call uq7777
-	ldd r24,Y+48
-	ldd r25,Y+49
-	subi r24,lo8(-(u))
-	sbci r25,hi8(-(u))
-	movw r2,r24
-	movw r26,r24
-	ld r24,X+
-	ld r25,X
+	movw r24,r2
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	movw r30,r24
+	subi r30,lo8(-(u))
+	sbci r31,hi8(-(u))
+	std Y+49,r31
+	std Y+48,r30
+	ld r24,Z
+	ldd r25,Z+1
 	movw r22,r10
 	call __udivmodhi4
-	movw r30,r16
-	ldd r24,Z+16
-	ldd r25,Z+17
+	movw r26,r16
+	adiw r26,16
+	ld r24,X+
+	ld r25,X
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	movw r22,r10
 	call __udivmodhi4
+	movw r30,r16
 	ldd r24,Z+22
 	ldd r25,Z+23
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
-	movw r22,r10
-	call __udivmodhi4
-	movw r26,r16
-	adiw r26,20
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
 	ld r24,X+
 	ld r25,X
+	movw r22,r10
+	call __udivmodhi4
+	ldd r24,Z+20
+	ldd r25,Z+21
 	cp r24,r22
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	movw r22,r10
 	call __udivmodhi4
-	movw r30,r16
 	ldd r24,Z+18
 	ldd r25,Z+19
 	cp r24,r22
@@ -3728,98 +3782,97 @@
 	movw r22,r12
 	movw r24,r14
 	call ur7777
-	movw r26,r2
-	ld r24,X+
-	ld r25,X
+	ldd r30,Y+48
+	ldd r31,Y+49
+	ld r24,Z
+	ldd r25,Z+1
 	movw r22,r10
 	call __udivmodhi4
-	movw r30,r16
-	ldd r18,Z+16
-	ldd r19,Z+17
+	movw r26,r16
+	adiw r26,16
+	ld r18,X+
+	ld r19,X
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	movw r26,r2
-	adiw r26,6
-	ld r24,X+
-	ld r25,X
+	ldd r24,Z+6
+	ldd r25,Z+7
 	movw r22,r10
 	call __udivmodhi4
+	movw r30,r16
 	ldd r18,Z+22
 	ldd r19,Z+23
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	movw r30,r2
-	ldd r24,Z+4
-	ldd r25,Z+5
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,4
+	ld r24,X+
+	ld r25,X
 	movw r22,r10
 	call __udivmodhi4
-	movw r26,r16
-	adiw r26,20
-	ld r18,X+
-	ld r19,X
+	ldd r18,Z+20
+	ldd r19,Z+21
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldd r24,Z+2
-	ldd r25,Z+3
+	ldd r26,Y+48
+	ldd r27,Y+49
+	adiw r26,2
+	ld r24,X+
+	ld r25,X
 	movw r22,r10
 	call __udivmodhi4
-	movw r30,r16
 	ldd r18,Z+18
 	ldd r19,Z+19
 	cp r18,r24
 	cpc r19,r25
 	breq .+2
 	rjmp .L42
-	ldi r31,16
-	add r12,r31
+	ldi r27,16
+	add r12,r27
 	adc r13,__zero_reg__
-	ldd r24,Y+50
-	ldd r25,Y+51
-	sbiw r24,1
+	dec r2
+	or r2,r3
 	breq .L61
-	ldi r25,lo8(1)
-	std Y+50,r25
-	std Y+51,__zero_reg__
+	clr r2
+	inc r2
+	mov r3,__zero_reg__
 	rjmp .L43
 .L61:
-	ldi r26,lo8(s)
-	mov r12,r26
-	ldi r26,hi8(s)
-	mov r13,r26
-	std Y+49,__zero_reg__
-	std Y+48,__zero_reg__
-	ldi r27,lo8(3)
-	mov r8,r27
+	ldi r20,lo8(s)
+	mov r12,r20
+	ldi r20,hi8(s)
+	mov r13,r20
+	mov r15,__zero_reg__
+	mov r14,__zero_reg__
+	ldi r21,lo8(3)
+	mov r8,r21
 	mov r9,__zero_reg__
-	ldi r24,lo8(6)
-	mov r6,r24
+	ldi r22,lo8(6)
+	mov r6,r22
 	mov r7,__zero_reg__
-	ldi r25,lo8(14)
-	mov r4,r25
+	ldi r23,lo8(14)
+	mov r4,r23
 	mov r5,__zero_reg__
-	ldi r18,lo8(7)
-	mov r10,r18
+	ldi r30,lo8(7)
+	mov r10,r30
 	mov r11,__zero_reg__
 .L58:
 	movw r22,r12
 	movw r24,r16
 	call sq4444
-	ldd r14,Y+48
-	ldd r15,Y+49
-	swap r14
-	swap r15
-	ldi r31,0xf0
-	and r15,r31
-	eor r15,r14
-	and r14,r31
-	eor r15,r14
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ld r24,Z
@@ -3839,6 +3892,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+6
@@ -3858,6 +3917,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+4
@@ -3878,6 +3943,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+2
@@ -3900,6 +3971,12 @@
 	movw r24,r16
 	call sr4444
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ld r18,Z
@@ -3922,6 +3999,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+6
@@ -3944,6 +4027,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+4
@@ -3967,6 +4056,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+2
@@ -3992,6 +4087,12 @@
 	movw r24,r16
 	call sq1428
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	movw r26,r16
@@ -4022,6 +4123,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+4
@@ -4040,6 +4147,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r24,Z+2
@@ -4068,6 +4181,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+6
@@ -4090,6 +4209,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+4
@@ -4113,6 +4238,12 @@
 	breq .+2
 	rjmp .L42
 	movw r30,r14
+	swap r30
+	swap r31
+	andi r31,0xf0
+	eor r31,r30
+	andi r30,0xf0
+	eor r31,r30
 	subi r30,lo8(-(s))
 	sbci r31,hi8(-(s))
 	ldd r18,Z+2
@@ -4138,10 +4269,17 @@
 	movw r24,r16
 	call sq3333
 	movw r24,r14
-	subi r24,lo8(-(s))
-	sbci r25,hi8(-(s))
-	movw r2,r24
-	movw r26,r24
+	swap r24
+	swap r25
+	andi r25,0xf0
+	eor r25,r24
+	andi r24,0xf0
+	eor r25,r24
+	movw r18,r24
+	subi r18,lo8(-(s))
+	sbci r19,hi8(-(s))
+	movw r2,r18
+	movw r26,r18
 	ld r24,X+
 	ld r25,X
 	movw r22,r8
@@ -4153,7 +4291,7 @@
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r26,r2
+	movw r26,r18
 	adiw r26,6
 	ld r24,X+
 	ld r25,X
@@ -4165,7 +4303,7 @@
 	cpc r25,r23
 	breq .+2
 	rjmp .L42
-	movw r30,r2
+	movw r30,r18
 	ldd r24,Z+4
 	ldd r25,Z+5
 	movw r22,r8
@@ -4547,19 +4685,19 @@
 	ldi r31,16
 	add r12,r31
 	adc r13,__zero_reg__
-	ldd r24,Y+48
-	ldd r25,Y+49
-	sbiw r24,1
+	ldi r18,1
+	cp r14,r18
+	cpc r15,__zero_reg__
 	breq .L62
-	ldi r25,lo8(1)
-	std Y+48,r25
-	std Y+49,__zero_reg__
+	clr r14
+	inc r14
+	mov r15,__zero_reg__
 	rjmp .L58
 .L62:
 	ldi r25,0
 	ldi r24,0
 /* epilogue start */
-	adiw r28,51
+	adiw r28,49
 	in __tmp_reg__,__SREG__
 	cli
 	out __SP_H__,r29
