DIR a828e71e7bdd8837dd73779c7ae8bc38
--- 980223.c.{atmega128}.{s}.{vanilla}.s	2020-08-07 12:13:40.806423734 +0000
+++ 980223.c.{atmega128}.{s}.{ccmode}.s	2020-08-07 12:13:41.538413871 +0000
@@ -147,18 +147,11 @@
 	brne 0b
 	ldi r24,lo8(nil)
 	ldi r25,hi8(nil)
-	mov r14,r24
-	std Y+8,r25
-	std Y+7,r24
-	std Y+9,__zero_reg__
-	std Y+10,__zero_reg__
-	std Y+11,__zero_reg__
-	std Y+12,__zero_reg__
-	ldd r15,Y+8
+	movw r14,r24
 	ldi r16,0
-	ldd r17,Y+10
-	ldd r18,Y+11
-	ldd r19,Y+12
+	ldi r17,0
+	ldi r18,0
+	ldi r19,0
 	ldd r20,Y+13
 	ldd r21,Y+14
 	ldd r22,Y+15
