DIR bc5bec41c97c0073199de5cb4ad8902f
--- pr49279.c.{atmega128}.{s}.{vanilla}.s	2020-08-05 19:05:33.814805476 +0000
+++ pr49279.c.{atmega128}.{s}.{lra}.s	2020-08-06 13:15:30.386557250 +0000
@@ -44,11 +44,9 @@
 	movw r24,r28
 	adiw r24,1
 	call bar
-	movw r26,r24
-	adiw r26,2+1
-	st X,r17
-	st -X,r16
-	sbiw r26,2
+	movw r30,r24
+	std Z+3,r17
+	std Z+2,r16
 	ldd r30,Y+3
 	ldd r31,Y+4
 	std Z+1,__zero_reg__
